---
title: "Create datasets for machine learning"
author: 
  - name: Levi Waldron
    affiliation:
    - CUNY Graduate School of Public Health and Health Policy
    email: levi.waldron@sph.cuny.edu
output: 
  BiocStyle::html_document:
    self_contained: yes
    toc: true
    toc_float: true
    toc_depth: 2
    code_folding: show
date: "`r BiocStyle::doc_date()`"
package: "`r BiocStyle::pkg_ver('curatedMetagenomicAnalyses')`"
vignette: >
  %\VignetteIndexEntry{ML datasets from cMD}
  %\VignetteEngine{knitr::rmarkdown}
  %\VignetteEncoding{UTF-8}  
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE)
```

This vignette identifies and writes to disk datasets for use in multi-dataset machine learning. For each relevant study condition, two files are written:

1. A `.rda` file containing a TreeSummarizedExperiment with full colData, rowData, a phylogenetic tree, and assay data containing taxonomic data.
2. A `.csv.gz` file containing only key sample metadata, and taxonomic data (variables in columns)

See [source file on GitHub](https://github.com/waldronlab/curatedMetagenomicAnalyses/blob/main/vignettes/MLdatasets.Rmd)

Do `BiocManager::install("waldronlab/curatedMetagenomicAnalyses")` to install the
`makeSEforCondition` function and to run this vignette.

Packages used here:

```{r, message=FALSE}
library(curatedMetagenomicData)
library(curatedMetagenomicAnalyses)
library(dplyr)
```

# Investigate potential response variables

These are the 10 study conditions most commonly found in curatedMetagenomicData:

```{r}
data("sampleMetadata")
availablediseases <- pull(sampleMetadata, study_condition) |>
  table() |>
  sort(decreasing = TRUE)
availablediseases
```

And the number of studies they are found in:
```{r}
studies <- lapply(names(availablediseases), function(x){
  filter(sampleMetadata, study_condition %in% x) |>
    pull(study_name) |>
    unique()
})
names(studies) <- names(availablediseases)
studies <- studies[-grep("control", names(studies))] #get rid of controls
studies <- studies[sapply(studies, length) > 1] #available in more than one study
studies
```

Each of these datasets has six data types associated with it; for example:
```{r}
curatedMetagenomicData("JieZ_2017.+")
```

# Write relative abundance datasets to disk

```{r writefiles, message=FALSE, cache=TRUE}
for (i in seq_along(studies)){
  cond <- names(studies)[i]
  se <-
    curatedMetagenomicAnalyses::makeSEforCondition(cond, removestudies = "HMP_2019_ibdmdb", dataType = "relative_abundance")
  print(paste("Next study condition:", cond, " /// Body site: ", unique(colData(se)$body_site)))
  print(with(colData(se), table(study_name, study_condition)))
  cat("\n \n")
  save(se, file = paste0(cond, ".rda"))
  flattext <- select(as.data.frame(colData(se)), c("study_name", "study_condition", "subject_id"))
  rownames(flattext) <- colData(se)$sample_id
  flattext <- cbind(flattext, data.frame(t(assay(se))))
  write.csv(flattext, file = paste0(cond, ".csv"))
  system(paste0("gzip ", cond, ".csv"))
}
```

# Direct link to files

Download the .csv and .rda files directly from https://www.dropbox.com/sh/0t0nbhj9eqm3wkq/AACZIw42WA-uHjzo97bG5tE6a?dl=0
