<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1.0">
<title>Exploration of the liver cirrhosis dataset â€¢ curatedMetagenomicAnalyses</title>
<!-- jquery --><script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.4.1/jquery.min.js" integrity="sha256-CSXorXvZcTkaix6Yvo6HppcZGetbYMGWSFlBw8HfCJo=" crossorigin="anonymous"></script><!-- Bootstrap --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/css/bootstrap.min.css" integrity="sha256-bZLfwXAP04zRMK2BjiO8iu9pf4FbLqX6zitd+tIvLhE=" crossorigin="anonymous">
<script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/js/bootstrap.min.js" integrity="sha256-nuL8/2cJ5NDSSwnKD8VqreErSWHtnEP9E7AySL+1ev4=" crossorigin="anonymous"></script><!-- bootstrap-toc --><link rel="stylesheet" href="../bootstrap-toc.css">
<script src="../bootstrap-toc.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous">
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous">
<!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- pkgdown --><link href="../pkgdown.css" rel="stylesheet">
<script src="../pkgdown.js"></script><meta property="og:title" content="Exploration of the liver cirrhosis dataset">
<meta property="og:description" content="curatedMetagenomicAnalyses">
<!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->
</head>
<body data-spy="scroll" data-target="#toc">
    <div class="container template-article">
      <header><div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../index.html">curatedMetagenomicAnalyses</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="Released version">0.3.0</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
<li>
  <a href="../index.html">Home</a>
</li>
<li>
  <a href="../reference/index.html">Reference</a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" aria-expanded="false">
    Analyses
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
<li class="dropdown-header">R Vignettes</li>
    <li>
      <a href="../articles/MLdatasets.html">Create datasets for machine learning</a>
    </li>
    <li>
      <a href="../articles/explorecirrhosis.html">Exploration of the liver cirrhosis dataset</a>
    </li>
    <li>
      <a href="../articles/Age_metaanalysis_vignette.html">Meta-analysis of age-related microbial species using cMD3</a>
    </li>
    <li>
      <a href="../articles/Sex_metaanalysis_vignette.html">Meta-analysis of sex-related microbial species using cMD3</a>
    </li>
    <li class="dropdown-header">Python Notebooks</li>
    <li>
      <a href="https://github.com/waldronlab/curatedMetagenomicAnalyses/blob/main/vignettes/curatedMetagenomicData%203%20CLI%20interface%2C%20sex-contrast%20microbiome%20meta-analysis.ipynb">Sex-related differences in the human microbiome using cMD3 and Python3</a>
    </li>
    <li class="dropdown-header">Supplementary Materials</li>
    <li>
      <a href="https://github.com/waldronlab/curatedMetagenomicAnalyses/blob/main/vignettes/installation.ipynb">Installing Python dependencies in Linux (Python Notebook)</a>
    </li>
  </ul>
</li>
      </ul>
<ul class="nav navbar-nav navbar-right">
<li>
  <a href="https://github.com/waldronlab/curatedMetagenomicAnalyses">
    <span class="fab fa-github fa-lg"></span>
     
  </a>
</li>
      </ul>
</div>
<!--/.nav-collapse -->
  </div>
<!--/.container -->
</div>
<!--/.navbar -->

      

      </header><script src="explorecirrhosis_files/header-attrs-2.9/header-attrs.js"></script><link href="explorecirrhosis_files/table1-1.0/table1_defaults.css" rel="stylesheet">
<div class="row">
  <div class="col-md-9 contents">
    <div class="page-header toc-ignore">
      <h1 data-toc-skip>Exploration of the liver cirrhosis dataset</h1>
                        <h4 class="author">Levi Waldron</h4>
            
            <h4 class="date">July 19, 2021</h4>
      
      <small class="dont-index">Source: <a href="https://github.com/waldronlab/curatedMetagenomicAnalyses/blob/master/vignettes/explorecirrhosis.Rmd"><code>vignettes/explorecirrhosis.Rmd</code></a></small>
      <div class="hidden name"><code>explorecirrhosis.Rmd</code></div>

    </div>

    
    
<p>Packages used here:</p>
<div class="sourceCode" id="cb1"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span><span class="op">(</span><span class="va"><a href="https://github.com/waldronlab/curatedMetagenomicData">curatedMetagenomicData</a></span><span class="op">)</span>
<span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span><span class="op">(</span><span class="va"><a href="https://github.com/waldronlab/curatedMetagenomicAnalyses">curatedMetagenomicAnalyses</a></span><span class="op">)</span>
<span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span><span class="op">(</span><span class="va"><a href="https://dplyr.tidyverse.org">dplyr</a></span><span class="op">)</span></code></pre></div>
<div id="cirrhosis-exploratory-analysis" class="section level1">
<h1 class="hasAnchor">
<a href="#cirrhosis-exploratory-analysis" class="anchor"></a>Cirrhosis exploratory analysis</h1>
<div class="sourceCode" id="cb2"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">se</span> <span class="op">&lt;-</span>
    <span class="fu">curatedMetagenomicAnalyses</span><span class="fu">::</span><span class="fu"><a href="../reference/makeSEforCondition.html">makeSEforCondition</a></span><span class="op">(</span><span class="st">"cirrhosis"</span>, removestudies <span class="op">=</span> <span class="st">"HMP_2019_ibdmdb"</span>, dataType <span class="op">=</span> <span class="st">"relative_abundance"</span><span class="op">)</span></code></pre></div>
<pre><code>## snapshotDate(): 2021-05-18</code></pre>
<pre><code>## 
## $`2021-03-31.LoombaR_2017.relative_abundance`
## dropping rows without rowTree matches:
##   k__Bacteria|p__Actinobacteria|c__Coriobacteriia|o__Coriobacteriales|f__Coriobacteriaceae|g__Collinsella|s__Collinsella_stercoris
##   k__Bacteria|p__Actinobacteria|c__Coriobacteriia|o__Coriobacteriales|f__Coriobacteriaceae|g__Enorma|s__[Collinsella]_massiliensis
##   k__Bacteria|p__Firmicutes|c__Bacilli|o__Bacillales|f__Bacillales_unclassified|g__Gemella|s__Gemella_bergeri
##   k__Bacteria|p__Firmicutes|c__Bacilli|o__Lactobacillales|f__Carnobacteriaceae|g__Granulicatella|s__Granulicatella_elegans
##   k__Bacteria|p__Firmicutes|c__Erysipelotrichia|o__Erysipelotrichales|f__Erysipelotrichaceae|g__Bulleidia|s__Bulleidia_extructa
##   k__Bacteria|p__Proteobacteria|c__Betaproteobacteria|o__Burkholderiales|f__Sutterellaceae|g__Sutterella|s__Sutterella_parvirubra
##   k__Bacteria|p__Synergistetes|c__Synergistia|o__Synergistales|f__Synergistaceae|g__Cloacibacillus|s__Cloacibacillus_evryensis</code></pre>
<pre><code>## $`2021-03-31.QinN_2014.relative_abundance`
## dropping rows without rowTree matches:
##   k__Bacteria|p__Actinobacteria|c__Coriobacteriia|o__Coriobacteriales|f__Atopobiaceae|g__Olsenella|s__Olsenella_profusa
##   k__Bacteria|p__Actinobacteria|c__Coriobacteriia|o__Coriobacteriales|f__Coriobacteriaceae|g__Collinsella|s__Collinsella_stercoris
##   k__Bacteria|p__Actinobacteria|c__Coriobacteriia|o__Coriobacteriales|f__Coriobacteriaceae|g__Enorma|s__[Collinsella]_massiliensis
##   k__Bacteria|p__Firmicutes|c__Bacilli|o__Bacillales|f__Bacillales_unclassified|g__Gemella|s__Gemella_bergeri
##   k__Bacteria|p__Firmicutes|c__Bacilli|o__Lactobacillales|f__Carnobacteriaceae|g__Granulicatella|s__Granulicatella_elegans
##   k__Bacteria|p__Firmicutes|c__Clostridia|o__Clostridiales|f__Ruminococcaceae|g__Ruminococcus|s__Ruminococcus_champanellensis
##   k__Bacteria|p__Firmicutes|c__Erysipelotrichia|o__Erysipelotrichales|f__Erysipelotrichaceae|g__Bulleidia|s__Bulleidia_extructa
##   k__Bacteria|p__Proteobacteria|c__Betaproteobacteria|o__Burkholderiales|f__Sutterellaceae|g__Sutterella|s__Sutterella_parvirubra
##   k__Bacteria|p__Synergistetes|c__Synergistia|o__Synergistales|f__Synergistaceae|g__Cloacibacillus|s__Cloacibacillus_evryensis</code></pre>
<div class="sourceCode" id="cb6"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span><span class="op">(</span><span class="va"><a href="https://github.com/benjaminrich/table1">table1</a></span><span class="op">)</span></code></pre></div>
<pre><code>## 
## Attaching package: 'table1'</code></pre>
<pre><code>## The following objects are masked from 'package:base':
## 
##     units, units&lt;-</code></pre>
<div class="sourceCode" id="cb9"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">df</span> <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/data.frame.html">data.frame</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/pkg/SummarizedExperiment/man/SummarizedExperiment-class.html">colData</a></span><span class="op">(</span><span class="va">se</span><span class="op">)</span><span class="op">[</span>, <span class="fl">1</span><span class="op">:</span><span class="fl">10</span><span class="op">]</span><span class="op">)</span>
<span class="fu"><a href="https://rdrr.io/pkg/table1/man/label.html">label</a></span><span class="op">(</span><span class="va">df</span><span class="op">$</span><span class="va">study_name</span><span class="op">)</span> <span class="op">=</span> <span class="st">"Study Name"</span>
<span class="fu"><a href="https://rdrr.io/pkg/table1/man/label.html">label</a></span><span class="op">(</span><span class="va">df</span><span class="op">$</span><span class="va">body_site</span><span class="op">)</span> <span class="op">=</span> <span class="st">"Body site"</span>
<span class="fu"><a href="https://rdrr.io/pkg/table1/man/label.html">label</a></span><span class="op">(</span><span class="va">df</span><span class="op">$</span><span class="va">study_condition</span><span class="op">)</span> <span class="op">=</span> <span class="st">"Study condition"</span>
<span class="fu"><a href="https://rdrr.io/pkg/table1/man/label.html">label</a></span><span class="op">(</span><span class="va">df</span><span class="op">$</span><span class="va">antibiotics_current_use</span><span class="op">)</span> <span class="op">=</span> <span class="st">"Current Antibiotics Use"</span>
<span class="fu"><a href="https://rdrr.io/pkg/table1/man/units.html">units</a></span><span class="op">(</span><span class="va">df</span><span class="op">$</span><span class="va">age</span><span class="op">)</span> <span class="op">=</span> <span class="st">"Years"</span></code></pre></div>
<div class="sourceCode" id="cb10"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="https://rdrr.io/pkg/table1/man/table1.html">table1</a></span><span class="op">(</span><span class="op">~</span> <span class="va">.</span> <span class="op">-</span>  <span class="va">subject_id</span> <span class="op">|</span> <span class="va">study_name</span>, data <span class="op">=</span> <span class="va">df</span><span class="op">)</span></code></pre></div>
<div class="Rtable1"><table class="table Rtable1">
<thead><tr>
<th class="rowlabel firstrow lastrow"></th>
<th class="firstrow lastrow"><span class="stratlabel">LoombaR_2017<br><span class="stratn">(N=45)</span></span></th>
<th class="firstrow lastrow"><span class="stratlabel">QinN_2014<br><span class="stratn">(N=237)</span></span></th>
<th class="firstrow lastrow"><span class="stratlabel">Overall<br><span class="stratn">(N=282)</span></span></th>
</tr></thead>
<tbody>
<tr>
<td class="rowlabel firstrow">Body site</td>
<td class="firstrow"></td>
<td class="firstrow"></td>
<td class="firstrow"></td>
</tr>
<tr>
<td class="rowlabel lastrow">stool</td>
<td class="lastrow">45 (100%)</td>
<td class="lastrow">237 (100%)</td>
<td class="lastrow">282 (100%)</td>
</tr>
<tr>
<td class="rowlabel firstrow">Current Antibiotics Use</td>
<td class="firstrow"></td>
<td class="firstrow"></td>
<td class="firstrow"></td>
</tr>
<tr>
<td class="rowlabel">no</td>
<td>0 (0%)</td>
<td>180 (75.9%)</td>
<td>180 (63.8%)</td>
</tr>
<tr>
<td class="rowlabel">yes</td>
<td>0 (0%)</td>
<td>57 (24.1%)</td>
<td>57 (20.2%)</td>
</tr>
<tr>
<td class="rowlabel lastrow">Missing</td>
<td class="lastrow">45 (100%)</td>
<td class="lastrow">0 (0%)</td>
<td class="lastrow">45 (16.0%)</td>
</tr>
<tr>
<td class="rowlabel firstrow">Study condition</td>
<td class="firstrow"></td>
<td class="firstrow"></td>
<td class="firstrow"></td>
</tr>
<tr>
<td class="rowlabel">cirrhosis</td>
<td>9 (20.0%)</td>
<td>123 (51.9%)</td>
<td>132 (46.8%)</td>
</tr>
<tr>
<td class="rowlabel lastrow">control</td>
<td class="lastrow">36 (80.0%)</td>
<td class="lastrow">114 (48.1%)</td>
<td class="lastrow">150 (53.2%)</td>
</tr>
<tr>
<td class="rowlabel firstrow">disease</td>
<td class="firstrow"></td>
<td class="firstrow"></td>
<td class="firstrow"></td>
</tr>
<tr>
<td class="rowlabel">cirrhosis</td>
<td>9 (20.0%)</td>
<td>9 (3.8%)</td>
<td>18 (6.4%)</td>
</tr>
<tr>
<td class="rowlabel">fatty_liver</td>
<td>36 (80.0%)</td>
<td>0 (0%)</td>
<td>36 (12.8%)</td>
</tr>
<tr>
<td class="rowlabel">ascites;cirrhosis</td>
<td>0 (0%)</td>
<td>9 (3.8%)</td>
<td>9 (3.2%)</td>
</tr>
<tr>
<td class="rowlabel">ascites;cirrhosis;hepatitis</td>
<td>0 (0%)</td>
<td>55 (23.2%)</td>
<td>55 (19.5%)</td>
</tr>
<tr>
<td class="rowlabel">ascites;cirrhosis;hepatitis;schistosoma</td>
<td>0 (0%)</td>
<td>3 (1.3%)</td>
<td>3 (1.1%)</td>
</tr>
<tr>
<td class="rowlabel">ascites;cirrhosis;hepatitis;wilson</td>
<td>0 (0%)</td>
<td>1 (0.4%)</td>
<td>1 (0.4%)</td>
</tr>
<tr>
<td class="rowlabel">ascites;cirrhosis;schistosoma</td>
<td>0 (0%)</td>
<td>1 (0.4%)</td>
<td>1 (0.4%)</td>
</tr>
<tr>
<td class="rowlabel">ascites;cirrhosis;wilson</td>
<td>0 (0%)</td>
<td>1 (0.4%)</td>
<td>1 (0.4%)</td>
</tr>
<tr>
<td class="rowlabel">cirrhosis;hepatitis</td>
<td>0 (0%)</td>
<td>43 (18.1%)</td>
<td>43 (15.2%)</td>
</tr>
<tr>
<td class="rowlabel">healthy</td>
<td>0 (0%)</td>
<td>114 (48.1%)</td>
<td>114 (40.4%)</td>
</tr>
<tr>
<td class="rowlabel lastrow">hepatitis</td>
<td class="lastrow">0 (0%)</td>
<td class="lastrow">1 (0.4%)</td>
<td class="lastrow">1 (0.4%)</td>
</tr>
<tr>
<td class="rowlabel firstrow">age (Years)</td>
<td class="firstrow"></td>
<td class="firstrow"></td>
<td class="firstrow"></td>
</tr>
<tr>
<td class="rowlabel">Mean (SD)</td>
<td>NA (NA)</td>
<td>46.5 (10.9)</td>
<td>46.5 (10.9)</td>
</tr>
<tr>
<td class="rowlabel">Median [Min, Max]</td>
<td>NA [NA, NA]</td>
<td>45.0 [18.0, 78.0]</td>
<td>45.0 [18.0, 78.0]</td>
</tr>
<tr>
<td class="rowlabel lastrow">Missing</td>
<td class="lastrow">45 (100%)</td>
<td class="lastrow">0 (0%)</td>
<td class="lastrow">45 (16.0%)</td>
</tr>
<tr>
<td class="rowlabel firstrow">age_category</td>
<td class="firstrow"></td>
<td class="firstrow"></td>
<td class="firstrow"></td>
</tr>
<tr>
<td class="rowlabel">adult</td>
<td>36 (80.0%)</td>
<td>221 (93.2%)</td>
<td>257 (91.1%)</td>
</tr>
<tr>
<td class="rowlabel">senior</td>
<td>9 (20.0%)</td>
<td>15 (6.3%)</td>
<td>24 (8.5%)</td>
</tr>
<tr>
<td class="rowlabel lastrow">schoolage</td>
<td class="lastrow">0 (0%)</td>
<td class="lastrow">1 (0.4%)</td>
<td class="lastrow">1 (0.4%)</td>
</tr>
<tr>
<td class="rowlabel firstrow">gender</td>
<td class="firstrow"></td>
<td class="firstrow"></td>
<td class="firstrow"></td>
</tr>
<tr>
<td class="rowlabel">female</td>
<td>0 (0%)</td>
<td>81 (34.2%)</td>
<td>81 (28.7%)</td>
</tr>
<tr>
<td class="rowlabel">male</td>
<td>0 (0%)</td>
<td>156 (65.8%)</td>
<td>156 (55.3%)</td>
</tr>
<tr>
<td class="rowlabel lastrow">Missing</td>
<td class="lastrow">45 (100%)</td>
<td class="lastrow">0 (0%)</td>
<td class="lastrow">45 (16.0%)</td>
</tr>
<tr>
<td class="rowlabel firstrow">country</td>
<td class="firstrow"></td>
<td class="firstrow"></td>
<td class="firstrow"></td>
</tr>
<tr>
<td class="rowlabel">USA</td>
<td>45 (100%)</td>
<td>0 (0%)</td>
<td>45 (16.0%)</td>
</tr>
<tr>
<td class="rowlabel lastrow">CHN</td>
<td class="lastrow">0 (0%)</td>
<td class="lastrow">237 (100%)</td>
<td class="lastrow">237 (84.0%)</td>
</tr>
</tbody>
</table></div>
<p>There seems to be strong correlation between study condition and current antibiotics use, raising questions of causality direction in these predictions:</p>
<div class="sourceCode" id="cb11"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="https://rdrr.io/r/base/table.html">table</a></span><span class="op">(</span><span class="va">df</span><span class="op">$</span><span class="va">study_condition</span>, <span class="va">df</span><span class="op">$</span><span class="va">antibiotics_current_use</span><span class="op">)</span></code></pre></div>
<pre><code>##            
##              no yes
##   cirrhosis  66  57
##   control   114   0</code></pre>
</div>
<div id="compositionality" class="section level1">
<h1 class="hasAnchor">
<a href="#compositionality" class="anchor"></a>Compositionality</h1>
<p>Many, but not all, columns add up to 100%:</p>
<div class="sourceCode" id="cb13"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="https://rdrr.io/r/base/summary.html">summary</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/colSums.html">colSums</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/pkg/SummarizedExperiment/man/SummarizedExperiment-class.html">assay</a></span><span class="op">(</span><span class="va">se</span><span class="op">)</span><span class="op">)</span><span class="op">)</span></code></pre></div>
<pre><code>##    Min. 1st Qu.  Median    Mean 3rd Qu.    Max. 
##   97.04   99.98  100.00   99.92  100.00  100.00</code></pre>
<p>The observation with minimum sum of relative abundances is:</p>
<div class="sourceCode" id="cb15"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="https://rdrr.io/pkg/SummarizedExperiment/man/SummarizedExperiment-class.html">colData</a></span><span class="op">(</span><span class="va">se</span><span class="op">)</span><span class="op">[</span><span class="fu"><a href="https://rdrr.io/r/base/which.min.html">which.min</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/colSums.html">colSums</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/pkg/SummarizedExperiment/man/SummarizedExperiment-class.html">assay</a></span><span class="op">(</span><span class="va">se</span><span class="op">)</span><span class="op">)</span><span class="op">)</span>, <span class="op">]</span></code></pre></div>
<pre><code>## DataFrame with 1 row and 31 columns
##       study_name  subject_id   body_site antibiotics_current_use
##      &lt;character&gt; &lt;character&gt; &lt;character&gt;             &lt;character&gt;
## LV-1   QinN_2014        LV-1       stool                      no
##      study_condition             disease       age age_category      gender
##          &lt;character&gt;         &lt;character&gt; &lt;integer&gt;  &lt;character&gt; &lt;character&gt;
## LV-1       cirrhosis cirrhosis;hepatitis        42        adult        male
##          country non_westernized sequencing_platform DNA_extraction_kit
##      &lt;character&gt;     &lt;character&gt;         &lt;character&gt;        &lt;character&gt;
## LV-1         CHN              no       IlluminaHiSeq                 NA
##             PMID number_reads number_bases minimum_read_length
##      &lt;character&gt;    &lt;integer&gt;    &lt;numeric&gt;           &lt;integer&gt;
## LV-1    25079328     34522116   3452211600                 100
##      median_read_length      NCBI_accession      curator       BMI
##               &lt;integer&gt;         &lt;character&gt;  &lt;character&gt; &lt;numeric&gt;
## LV-1                100 ERR528291;ERR528292 Paolo_Manghi     19.03
##      antibiotics_family disease_subtype  creatine  albumine     alcohol
##             &lt;character&gt;     &lt;character&gt; &lt;numeric&gt; &lt;numeric&gt; &lt;character&gt;
## LV-1                 NA   HBV;cirrhosis        65      31.2         yes
##      disease_stage bilubirin prothrombin_time       inr       ctp
##          &lt;integer&gt; &lt;numeric&gt;        &lt;numeric&gt; &lt;numeric&gt; &lt;integer&gt;
## LV-1            NA      1.82               NA      1.18         6</code></pre>
<p>Note, this a result of species where phylogenetic information was unavailable in building of the <code>TreeSummarizedExperiment</code>. It would be possible to take data directly from <code>ExperimentHub</code> to avoid the data loss, but the data loss is rare and small and probably unimportant for most purposes.</p>
</div>
<div id="siamcat-analysis" class="section level1">
<h1 class="hasAnchor">
<a href="#siamcat-analysis" class="anchor"></a>SIAMCAT analysis</h1>
<p>From the <a href="https://www.bioconductor.org/packages/release/bioc/vignettes/SIAMCAT/inst/doc/SIAMCAT_vignette.html">SIAMCAT vignette</a>.</p>
<div class="sourceCode" id="cb17"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="kw">if</span><span class="op">(</span><span class="op">!</span><span class="fu"><a href="https://rdrr.io/r/base/ns-load.html">requireNamespace</a></span><span class="op">(</span><span class="st">"SIAMCAT"</span><span class="op">)</span><span class="op">)</span>
  <span class="fu">BiocManager</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/BiocManager/man/install.html">install</a></span><span class="op">(</span><span class="st">"SIAMCAT"</span><span class="op">)</span>
<span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span><span class="op">(</span><span class="va">SIAMCAT</span><span class="op">)</span></code></pre></div>
<div class="sourceCode" id="cb18"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">labs</span> <span class="op">&lt;-</span>
  <span class="fu"><a href="https://rdrr.io/pkg/SIAMCAT/man/create.label.html">create.label</a></span><span class="op">(</span>meta <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/data.frame.html">data.frame</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/pkg/SummarizedExperiment/man/SummarizedExperiment-class.html">colData</a></span><span class="op">(</span><span class="va">se</span><span class="op">)</span><span class="op">)</span>,
               label <span class="op">=</span> <span class="st">'study_condition'</span>,
               case <span class="op">=</span> <span class="st">'cirrhosis'</span><span class="op">)</span></code></pre></div>
<pre><code>## Label used as case:
##    cirrhosis
## Label used as control:
##    control</code></pre>
<pre><code>## + finished create.label.from.metadata in 0.013 s</code></pre>
<div class="sourceCode" id="cb21"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">sc.obj</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/SIAMCAT/man/siamcat.html">siamcat</a></span><span class="op">(</span>feat<span class="op">=</span><span class="fu"><a href="https://rdrr.io/pkg/SummarizedExperiment/man/SummarizedExperiment-class.html">assay</a></span><span class="op">(</span><span class="va">se</span><span class="op">)</span><span class="op">/</span><span class="fl">100</span>,
    label<span class="op">=</span><span class="va">labs</span>,
    meta<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/data.frame.html">data.frame</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/pkg/SummarizedExperiment/man/SummarizedExperiment-class.html">colData</a></span><span class="op">(</span><span class="va">se</span><span class="op">)</span><span class="op">)</span><span class="op">)</span></code></pre></div>
<pre><code>## + starting validate.data</code></pre>
<pre><code>## +++ checking overlap between labels and features</code></pre>
<pre><code>## + Keeping labels of 282 sample(s).</code></pre>
<pre><code>## +++ checking sample number per class</code></pre>
<pre><code>## +++ checking overlap between samples and metadata</code></pre>
<pre><code>## + finished validate.data in 0.07 s</code></pre>
<div class="sourceCode" id="cb28"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu">show</span><span class="op">(</span><span class="va">sc.obj</span><span class="op">)</span></code></pre></div>
<pre><code>## siamcat-class object
## label()                Label object:         150 control and 132 cirrhosis samples
## 
## contains phyloseq-class experiment-level object @phyloseq:
## phyloseq@otu_table()   OTU Table:            [ 693 taxa and 282 samples ]
## phyloseq@sam_data()    Sample Data:          [ 282 samples by 31 sample variables ]</code></pre>
<p>Unsupervised filtering:</p>
<div class="sourceCode" id="cb30"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">sc.obj</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/SIAMCAT/man/filter.features.html">filter.features</a></span><span class="op">(</span><span class="va">sc.obj</span>,
    filter.method <span class="op">=</span> <span class="st">'abundance'</span>,
    cutoff <span class="op">=</span> <span class="fl">0.001</span><span class="op">)</span></code></pre></div>
<pre><code>## Features successfully filtered</code></pre>
<div id="association-testing" class="section level2">
<h2 class="hasAnchor">
<a href="#association-testing" class="anchor"></a>Association testing</h2>
<div class="sourceCode" id="cb32"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">sc.obj</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/SIAMCAT/man/check.associations.html">check.associations</a></span><span class="op">(</span>
    <span class="va">sc.obj</span>,
    sort.by <span class="op">=</span> <span class="st">'fc'</span>,
    alpha <span class="op">=</span> <span class="fl">0.05</span>,
    mult.corr <span class="op">=</span> <span class="st">"fdr"</span>,
    detect.lim <span class="op">=</span> <span class="fl">10</span> <span class="op">^</span><span class="op">-</span><span class="fl">6</span>,
    plot.type <span class="op">=</span> <span class="st">"quantile.box"</span>,
    panels <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="st">"fc"</span>, <span class="st">"prevalence"</span>, <span class="st">"auroc"</span><span class="op">)</span><span class="op">)</span></code></pre></div>
<pre><code>## ### WARNING: Not plotting to a pdf-file.
## ### The plot is optimized for landscape DIN-A4 (or similar) layout.
## ### Please make sure that your plotting region is large enough!!!
## ### Use at your own risk...</code></pre>
<pre><code>## Are you sure that you want to continue? (Yes/no/cancel)</code></pre>
<p><img src="explorecirrhosis_files/figure-html/unnamed-chunk-11-1.png" width="700"></p>
</div>
</div>
  </div>

  <div class="col-md-3 hidden-xs hidden-sm" id="pkgdown-sidebar">

        <nav id="toc" data-toggle="toc"><h2 data-toc-skip>Contents</h2>
    </nav>
</div>

</div>



      <footer><div class="copyright">
  <p>Developed by Levi Waldron, Paolo Manghi, Davide Galzato, Jennifer Wokaty.</p>
</div>

<div class="pkgdown">
  <p>Site built with <a href="https://pkgdown.r-lib.org/">pkgdown</a> 1.6.1.</p>
</div>

      </footer>
</div>

  


  </body>
</html>
