<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1.0">
<title>List all species present in colorectal cancer datasets â€¢ curatedMetagenomicAnalyses</title>
<!-- jquery --><script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.4.1/jquery.min.js" integrity="sha256-CSXorXvZcTkaix6Yvo6HppcZGetbYMGWSFlBw8HfCJo=" crossorigin="anonymous"></script><!-- Bootstrap --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/css/bootstrap.min.css" integrity="sha256-bZLfwXAP04zRMK2BjiO8iu9pf4FbLqX6zitd+tIvLhE=" crossorigin="anonymous">
<script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/js/bootstrap.min.js" integrity="sha256-nuL8/2cJ5NDSSwnKD8VqreErSWHtnEP9E7AySL+1ev4=" crossorigin="anonymous"></script><!-- bootstrap-toc --><link rel="stylesheet" href="../bootstrap-toc.css">
<script src="../bootstrap-toc.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous">
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous">
<!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- pkgdown --><link href="../pkgdown.css" rel="stylesheet">
<script src="../pkgdown.js"></script><meta property="og:title" content="List all species present in colorectal cancer datasets">
<meta property="og:description" content="curatedMetagenomicAnalyses">
<!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->
</head>
<body data-spy="scroll" data-target="#toc">
    

    <div class="container template-article">
      <header><div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../index.html">curatedMetagenomicAnalyses</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="">0.4.0</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
<li>
  <a href="../index.html">Home</a>
</li>
<li>
  <a href="../reference/index.html">Reference</a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" data-bs-toggle="dropdown" aria-expanded="false">
    Analyses
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
<li class="dropdown-header">R Vignettes</li>
    <li>
      <a href="../articles/MLdatasets.html">Create datasets for machine learning</a>
    </li>
    <li>
      <a href="../articles/explorecirrhosis.html">Exploration of the liver cirrhosis dataset</a>
    </li>
    <li>
      <a href="../articles/identify_CRC_species.html">Prevalences of all species found in stool microbiomes of colorectal cancer patients (dynamic table)</a>
    </li>
    <li>
      <a href="../articles/Age_metaanalysis_vignette.html">Meta-analysis of age-related microbial species using cMD3</a>
    </li>
    <li>
      <a href="../articles/Sex_metaanalysis_vignette.html">Meta-analysis of sex-related microbial species using cMD3</a>
    </li>
    <li>
      <a href="../articles/NUI-Galway-Metagenomics-Workshop.html">NUI Galway Metagenomics Workshop</a>
    </li>
    <li class="dropdown-header">Python Notebooks</li>
    <li>
      <a href="https://github.com/waldronlab/curatedMetagenomicAnalyses/blob/main/vignettes/sexContrastMicrobiomeAnalysis.ipynb">Sex-related differences in the human microbiome using cMD3 and Python3</a>
    </li>
    <li class="dropdown-header">Supplementary Materials</li>
    <li>
      <a href="https://github.com/waldronlab/curatedMetagenomicAnalyses/blob/main/vignettes/installation.ipynb">Installing Python dependencies in Linux (Python Notebook)</a>
    </li>
  </ul>
</li>
      </ul>
<ul class="nav navbar-nav navbar-right">
<li>
  <a href="https://github.com/waldronlab/curatedMetagenomicAnalyses">
    <span class="fab fa-github fa-lg"></span>
     
  </a>
</li>
      </ul>
</div>
<!--/.nav-collapse -->
  </div>
<!--/.container -->
</div>
<!--/.navbar -->

      

      </header><link href="identify_CRC_species_files/table1-1.0/table1_defaults.css" rel="stylesheet">
<script src="identify_CRC_species_files/htmlwidgets-1.5.4/htmlwidgets.js"></script><link href="identify_CRC_species_files/datatables-css-0.0.0/datatables-crosstalk.css" rel="stylesheet">
<script src="identify_CRC_species_files/datatables-binding-0.22/datatables.js"></script><link href="identify_CRC_species_files/dt-core-1.11.3/css/jquery.dataTables.min.css" rel="stylesheet">
<link href="identify_CRC_species_files/dt-core-1.11.3/css/jquery.dataTables.extra.css" rel="stylesheet">
<script src="identify_CRC_species_files/dt-core-1.11.3/js/jquery.dataTables.min.js"></script><link href="identify_CRC_species_files/crosstalk-1.2.0/css/crosstalk.min.css" rel="stylesheet">
<script src="identify_CRC_species_files/crosstalk-1.2.0/js/crosstalk.min.js"></script><div class="row">
  <div class="col-md-9 contents">
    <div class="page-header toc-ignore">
      <h1 data-toc-skip>List all species present in colorectal cancer
datasets</h1>
                        <h4 data-toc-skip class="author">Levi
Waldron</h4>
            <address class="author_afil">
      CUNY Graduate School of Public Health and Health
Policy<br><a class="author_email" href="mailto:#"></a><a href="mailto:levi.waldron@sph.cuny.edu" class="email">levi.waldron@sph.cuny.edu</a>
      </address>
                  
            <h4 data-toc-skip class="date">6 April 2022</h4>
      
      <small class="dont-index">Source: <a href="https://github.com/waldronlab/curatedMetagenomicAnalyses/blob/HEAD/vignettes/identify_CRC_species.Rmd"><code>vignettes/identify_CRC_species.Rmd</code></a></small>
      <div class="hidden name"><code>identify_CRC_species.Rmd</code></div>

    </div>

    
    
<div class="section level2">
<h2 id="overview">Overview<a class="anchor" aria-label="anchor" href="#overview"></a>
</h2>
<p>This analysis: 1. selects a subset of samples (all CRC-related in
this example), 2. uses the <em><a href="https://CRAN.R-project.org/package=table1" class="external-link">table1</a></em> package
to display a table of the characteristics of the included cohort, 3.
sorts species in order of descending prevalence, 4. uses the <em><a href="https://CRAN.R-project.org/package=DT" class="external-link">DT</a></em> package
<code>datatable</code> function to display a searchable, paged table of
prevalences, and 4. writes the prevalences to file.</p>
<p>Required packages:</p>
<div class="sourceCode" id="cb1"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="https://rdrr.io/r/base/message.html" class="external-link">suppressPackageStartupMessages</a></span><span class="op">(</span><span class="op">{</span>
  <span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://github.com/benjaminrich/table1" class="external-link">table1</a></span><span class="op">)</span>
  <span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://github.com/rstudio/DT" class="external-link">DT</a></span><span class="op">)</span>
  <span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://github.com/waldronlab/curatedMetagenomicData" class="external-link">curatedMetagenomicData</a></span><span class="op">)</span>
  <span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://dplyr.tidyverse.org" class="external-link">dplyr</a></span><span class="op">)</span>
<span class="op">}</span><span class="op">)</span></code></pre></div>
</div>
<div class="section level2">
<h2 id="select-samples">Select samples<a class="anchor" aria-label="anchor" href="#select-samples"></a>
</h2>
<p>Note that a few species without phylogenetic information are
lost.</p>
<div class="sourceCode" id="cb2"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">crc_subset</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://dplyr.tidyverse.org/reference/filter.html" class="external-link">filter</a></span><span class="op">(</span><span class="va">sampleMetadata</span>, <span class="va">study_condition</span> <span class="op">==</span> <span class="st">"CRC"</span><span class="op">)</span> <span class="op"><a href="https://rdrr.io/pkg/DT/man/DT-imports.html" class="external-link">%&gt;%</a></span>
  <span class="fu"><a href="https://rdrr.io/pkg/curatedMetagenomicData/man/returnSamples.html" class="external-link">returnSamples</a></span><span class="op">(</span>dataType <span class="op">=</span> <span class="st">"relative_abundance"</span>,
                rownames <span class="op">=</span> <span class="st">"short"</span><span class="op">)</span></code></pre></div>
</div>
<div class="section level2">
<h2 id="cohort-characteristics">Cohort characteristics<a class="anchor" aria-label="anchor" href="#cohort-characteristics"></a>
</h2>
<p>Create a summary table of the participants in this cohort:</p>
<div class="sourceCode" id="cb3"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu">table1</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/table1/man/table1.html" class="external-link">table1</a></span><span class="op">(</span> <span class="op">~</span> <span class="va">disease</span> <span class="op">+</span> <span class="va">disease_subtype</span> <span class="op">+</span> <span class="va">age</span> <span class="op">+</span> <span class="va">gender</span> <span class="op">+</span> <span class="va">country</span> <span class="op">+</span> <span class="va">study_name</span>,
                data <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/pkg/SummarizedExperiment/man/SummarizedExperiment-class.html" class="external-link">colData</a></span><span class="op">(</span><span class="va">crc_subset</span><span class="op">)</span><span class="op">)</span></code></pre></div>
<div class="Rtable1"><table class="table Rtable1">
<thead><tr>
<th class="rowlabel firstrow lastrow"></th>
<th class="firstrow lastrow"><span class="stratlabel">Overall<br><span class="stratn">(N=702)</span></span></th>
</tr></thead>
<tbody>
<tr>
<td class="rowlabel firstrow">disease</td>
<td class="firstrow"></td>
</tr>
<tr>
<td class="rowlabel">CRC</td>
<td>626 (89.2%)</td>
</tr>
<tr>
<td class="rowlabel">CRC;cholesterolemia</td>
<td>1 (0.1%)</td>
</tr>
<tr>
<td class="rowlabel">CRC;fatty_liver</td>
<td>3 (0.4%)</td>
</tr>
<tr>
<td class="rowlabel">CRC;fatty_liver;hypertension</td>
<td>12 (1.7%)</td>
</tr>
<tr>
<td class="rowlabel">CRC;hypercholesterolemia</td>
<td>3 (0.4%)</td>
</tr>
<tr>
<td class="rowlabel">CRC;hypercholesterolemia;hypertension</td>
<td>1 (0.1%)</td>
</tr>
<tr>
<td class="rowlabel">CRC;hypertension</td>
<td>20 (2.8%)</td>
</tr>
<tr>
<td class="rowlabel">CRC;metastases</td>
<td>1 (0.1%)</td>
</tr>
<tr>
<td class="rowlabel">CRC;T2D</td>
<td>29 (4.1%)</td>
</tr>
<tr>
<td class="rowlabel">CRC;T2D;fatty_liver;hypertension</td>
<td>4 (0.6%)</td>
</tr>
<tr>
<td class="rowlabel lastrow">CRC;T2D;hypertension</td>
<td class="lastrow">2 (0.3%)</td>
</tr>
<tr>
<td class="rowlabel firstrow">disease_subtype</td>
<td class="firstrow"></td>
</tr>
<tr>
<td class="rowlabel">adenocarcinoma</td>
<td>59 (8.4%)</td>
</tr>
<tr>
<td class="rowlabel">carcinoma</td>
<td>174 (24.8%)</td>
</tr>
<tr>
<td class="rowlabel lastrow">Missing</td>
<td class="lastrow">469 (66.8%)</td>
</tr>
<tr>
<td class="rowlabel firstrow">age</td>
<td class="firstrow"></td>
</tr>
<tr>
<td class="rowlabel">Mean (SD)</td>
<td>63.3 (11.0)</td>
</tr>
<tr>
<td class="rowlabel lastrow">Median [Min, Max]</td>
<td class="lastrow">64.0 [28.0, 90.0]</td>
</tr>
<tr>
<td class="rowlabel firstrow">gender</td>
<td class="firstrow"></td>
</tr>
<tr>
<td class="rowlabel">female</td>
<td>258 (36.8%)</td>
</tr>
<tr>
<td class="rowlabel lastrow">male</td>
<td class="lastrow">444 (63.2%)</td>
</tr>
<tr>
<td class="rowlabel firstrow">country</td>
<td class="firstrow"></td>
</tr>
<tr>
<td class="rowlabel">AUT</td>
<td>46 (6.6%)</td>
</tr>
<tr>
<td class="rowlabel">CAN</td>
<td>2 (0.3%)</td>
</tr>
<tr>
<td class="rowlabel">CHN</td>
<td>75 (10.7%)</td>
</tr>
<tr>
<td class="rowlabel">DEU</td>
<td>60 (8.5%)</td>
</tr>
<tr>
<td class="rowlabel">FRA</td>
<td>53 (7.5%)</td>
</tr>
<tr>
<td class="rowlabel">IND</td>
<td>30 (4.3%)</td>
</tr>
<tr>
<td class="rowlabel">ITA</td>
<td>61 (8.7%)</td>
</tr>
<tr>
<td class="rowlabel">JPN</td>
<td>298 (42.5%)</td>
</tr>
<tr>
<td class="rowlabel lastrow">USA</td>
<td class="lastrow">77 (11.0%)</td>
</tr>
<tr>
<td class="rowlabel firstrow">study_name</td>
<td class="firstrow"></td>
</tr>
<tr>
<td class="rowlabel">FengQ_2015</td>
<td>46 (6.6%)</td>
</tr>
<tr>
<td class="rowlabel">GuptaA_2019</td>
<td>30 (4.3%)</td>
</tr>
<tr>
<td class="rowlabel">HanniganGD_2017</td>
<td>27 (3.8%)</td>
</tr>
<tr>
<td class="rowlabel">ThomasAM_2018a</td>
<td>29 (4.1%)</td>
</tr>
<tr>
<td class="rowlabel">ThomasAM_2018b</td>
<td>32 (4.6%)</td>
</tr>
<tr>
<td class="rowlabel">ThomasAM_2019_c</td>
<td>40 (5.7%)</td>
</tr>
<tr>
<td class="rowlabel">VogtmannE_2016</td>
<td>52 (7.4%)</td>
</tr>
<tr>
<td class="rowlabel">WirbelJ_2018</td>
<td>60 (8.5%)</td>
</tr>
<tr>
<td class="rowlabel">YachidaS_2019</td>
<td>258 (36.8%)</td>
</tr>
<tr>
<td class="rowlabel">YuJ_2015</td>
<td>75 (10.7%)</td>
</tr>
<tr>
<td class="rowlabel lastrow">ZellerG_2014</td>
<td class="lastrow">53 (7.5%)</td>
</tr>
</tbody>
</table></div>
</div>
<div class="section level2">
<h2 id="show-species-in-order-of-decreasing-prevalence">Show species in order of decreasing prevalence<a class="anchor" aria-label="anchor" href="#show-species-in-order-of-decreasing-prevalence"></a>
</h2>
<p>Prevalences shown are the fraction of specimens from CRC patients
with non-zero relative abundance.</p>
<div class="sourceCode" id="cb4"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">prevalences</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/Matrix/man/colSums.html" class="external-link">rowSums</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/pkg/SummarizedExperiment/man/SummarizedExperiment-class.html" class="external-link">assay</a></span><span class="op">(</span><span class="va">crc_subset</span><span class="op">)</span> <span class="op">&gt;</span> <span class="fl">0</span><span class="op">)</span> <span class="op">/</span> <span class="fu"><a href="https://rdrr.io/r/base/nrow.html" class="external-link">ncol</a></span><span class="op">(</span><span class="va">crc_subset</span><span class="op">)</span> 
<span class="va">prevalences</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://tibble.tidyverse.org/reference/tibble.html" class="external-link">tibble</a></span><span class="op">(</span>species <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/names.html" class="external-link">names</a></span><span class="op">(</span><span class="va">prevalences</span><span class="op">)</span>, prevalence <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/Round.html" class="external-link">signif</a></span><span class="op">(</span><span class="va">prevalences</span>, <span class="fl">2</span><span class="op">)</span><span class="op">)</span> <span class="op"><a href="https://rdrr.io/pkg/DT/man/DT-imports.html" class="external-link">%&gt;%</a></span>
  <span class="fu"><a href="https://dplyr.tidyverse.org/reference/filter.html" class="external-link">filter</a></span><span class="op">(</span><span class="va">prevalence</span> <span class="op">&gt;</span> <span class="fl">0</span><span class="op">)</span> <span class="op"><a href="https://rdrr.io/pkg/DT/man/DT-imports.html" class="external-link">%&gt;%</a></span>
  <span class="fu"><a href="https://dplyr.tidyverse.org/reference/arrange.html" class="external-link">arrange</a></span><span class="op">(</span><span class="op">-</span><span class="va">prevalence</span><span class="op">)</span>
<span class="fu">DT</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/DT/man/datatable.html" class="external-link">datatable</a></span><span class="op">(</span><span class="va">prevalences</span><span class="op">)</span></code></pre></div>
<div id="htmlwidget-ab8a64999fab4364fc19" style="width:100%;height:auto;" class="datatables html-widget"></div>
<script type="application/json" data-for="htmlwidget-ab8a64999fab4364fc19">{"x":{"filter":"none","vertical":false,"data":[["1","2","3","4","5","6","7","8","9","10","11","12","13","14","15","16","17","18","19","20","21","22","23","24","25","26","27","28","29","30","31","32","33","34","35","36","37","38","39","40","41","42","43","44","45","46","47","48","49","50","51","52","53","54","55","56","57","58","59","60","61","62","63","64","65","66","67","68","69","70","71","72","73","74","75","76","77","78","79","80","81","82","83","84","85","86","87","88","89","90","91","92","93","94","95","96","97","98","99","100","101","102","103","104","105","106","107","108","109","110","111","112","113","114","115","116","117","118","119","120","121","122","123","124","125","126","127","128","129","130","131","132","133","134","135","136","137","138","139","140","141","142","143","144","145","146","147","148","149","150","151","152","153","154","155","156","157","158","159","160","161","162","163","164","165","166","167","168","169","170","171","172","173","174","175","176","177","178","179","180","181","182","183","184","185","186","187","188","189","190","191","192","193","194","195","196","197","198","199","200","201","202","203","204","205","206","207","208","209","210","211","212","213","214","215","216","217","218","219","220","221","222","223","224","225","226","227","228","229","230","231","232","233","234","235","236","237","238","239","240","241","242","243","244","245","246","247","248","249","250","251","252","253","254","255","256","257","258","259","260","261","262","263","264","265","266","267","268","269","270","271","272","273","274","275","276","277","278","279","280","281","282","283","284","285","286","287","288","289","290","291","292","293","294","295","296","297","298","299","300","301","302","303","304","305","306","307","308","309","310","311","312","313","314","315","316","317","318","319","320","321","322","323","324","325","326","327","328","329","330","331","332","333","334","335","336","337","338","339","340","341","342","343","344","345","346","347","348","349","350","351","352","353","354","355","356","357","358","359","360","361","362","363","364","365","366","367","368","369","370","371","372","373","374","375","376","377","378","379","380","381","382","383","384","385","386","387","388","389","390","391","392","393","394","395","396","397","398","399","400","401","402","403","404","405","406","407","408","409","410","411","412","413","414","415","416","417","418","419","420","421","422","423","424","425","426","427","428","429","430","431","432","433","434","435","436","437","438","439","440","441","442","443","444","445","446","447","448","449","450","451","452","453","454","455","456","457","458","459","460","461","462","463","464","465","466","467","468","469","470","471","472","473","474","475","476","477","478","479","480","481","482","483","484","485","486","487","488","489","490","491","492","493","494","495","496","497","498","499","500","501","502","503","504","505","506","507","508","509","510","511","512","513","514","515","516","517","518","519","520","521","522","523","524","525","526","527","528","529","530","531","532","533","534","535","536","537","538","539","540","541","542","543","544","545","546","547","548","549","550","551","552","553","554","555","556","557","558","559","560","561","562","563","564","565","566","567","568","569","570","571","572","573","574","575","576","577","578","579","580","581","582","583","584","585","586","587","588","589","590","591","592","593","594","595","596","597","598","599","600","601","602","603","604","605","606","607","608","609","610","611","612","613","614","615","616","617","618","619","620","621","622","623","624","625","626","627","628","629","630","631","632","633","634","635","636","637","638","639","640","641","642","643","644","645","646","647","648","649","650","651","652","653","654","655","656","657","658","659","660","661","662","663","664","665","666","667","668","669","670","671","672","673","674","675","676","677","678","679","680","681","682","683","684","685","686","687","688","689","690","691","692","693","694","695","696","697","698","699","700","701","702","703","704","705","706","707","708","709","710","711","712","713","714","715","716","717","718","719","720","721","722","723","724","725","726","727","728","729","730","731","732","733","734","735","736","737","738","739","740","741","742","743","744","745","746","747","748","749","750","751","752","753","754","755","756","757","758","759","760","761","762","763","764","765","766","767","768","769","770","771","772","773","774","775","776","777","778","779","780","781","782","783","784","785","786","787","788","789","790","791","792","793","794","795","796","797","798","799","800","801","802","803","804","805","806","807","808","809","810","811","812","813","814","815","816","817","818","819","820","821","822","823","824","825","826","827","828","829","830","831"],["Faecalibacterium prausnitzii","Bacteroides uniformis","Ruthenibacterium lactatiformans","Flavonifractor plautii","Phocaeicola vulgatus","Parabacteroides distasonis","Collinsella aerofaciens","Blautia wexlerae","Agathobaculum butyriciproducens","Bacteroides thetaiotaomicron","Fusicatenibacter saccharivorans","Anaerostipes hadrus","Bacteroides ovatus","Streptococcus salivarius","Roseburia inulinivorans","Blautia obeum","Bifidobacterium longum","Anaerobutyricum hallii","Streptococcus parasanguinis","Dorea longicatena","[Eubacterium] rectale","Dorea formicigenerans","Escherichia coli","[Ruminococcus] gnavus","Gordonibacter pamelaeae","Odoribacter splanchnicus","Lawsonibacter asaccharolyticus","Bacteroides xylanisolvens","Parabacteroides merdae","Alistipes putredinis","Roseburia hominis","Gemmiger formicilis","Bilophila wadsworthia","Alistipes finegoldii","Alistipes shahii","Eggerthella lenta","Coprococcus comes","Enterocloster bolteae","Bacteroides caccae","[Clostridium] leptum","Firmicutes bacterium CAG:83","Lachnospira eligens","Phocaeicola dorei","[Ruminococcus] torques","Coprococcus catus","Ruminococcus bromii","Roseburia faecis","Bacteroides stercoris","Eubacterium ramulus","Butyricimonas virosa","Bacteroides fragilis","Adlercreutzia equolifaciens","Bifidobacterium adolescentis","Enterocloster citroniae","Barnesiella intestinihominis","Akkermansia muciniphila","Roseburia intestinalis","Oscillibacter sp. 57_20","Hungatella hathewayi","[Clostridium] symbiosum","Eubacterium ventriosum","Intestinimonas butyriciproducens","Streptococcus mitis","Veillonella parvula","Gemella sanguinis","Bacteroides cellulosilyticus","Oscillibacter sp. CAG:241","Erysipelatoclostridium ramosum","Prevotella copri","Eubacterium sp. CAG:38","Holdemania filiformis","Anaerotruncus colihominis","Haemophilus parainfluenzae","Collinsella intestinalis","[Eubacterium] siraeum","Schaalia odontolytica","Streptococcus oralis","Ruminococcus bicirculans","[Clostridium] innocuum","Eubacterium sp. CAG:180","Clostridium sp. CAG:58","Lachnospira pectinoschiza","Bifidobacterium pseudocatenulatum","Alistipes indistinctus","Mogibacterium diversum","Parvimonas micra","Phocaeicola plebeius","Streptococcus thermophilus","Parasutterella excrementihominis","Eisenbergiella tayi","Phascolarctobacterium faecium","Butyricimonas synergistica","Veillonella atypica","Intestinibacter bartlettii","Slackia isoflavoniconvertens","Firmicutes bacterium CAG:145","Anaeromassilibacillus sp. An250","Rothia mucilaginosa","Roseburia sp. CAG:471","Veillonella dispar","Streptococcus infantis","Gemella morbillorum","Bacteroides eggerthii","Paraprevotella xylaniphila","Phocaeicola massiliensis","[Ruminococcus] lactaris","Phascolarctobacterium succinatutens","Enorma massiliensis","Methanobrevibacter smithii","Adlercreutzia caecimuris","Parabacteroides goldsteinii","Blautia sp. CAG:257","Enterocloster lavalensis","Holdemanella biformis","Actinomyces sp. ICM47","Enterocloster asparagiformis","Gemella haemolysans","Streptococcus gordonii","Enterocloster clostridioformis","Firmicutes bacterium CAG:94","Firmicutes bacterium CAG:110","Streptococcus vestibularis","Coprobacter fastidiosus","Peptostreptococcus stomatis","Clostridium sp. CAG:242","Streptococcus sanguinis","Proteobacteria bacterium CAG:139","Sellimonas intestinalis","Bacteroides finegoldii","Solobacterium moorei","Streptococcus australis","Actinomyces sp. HMSC035G02","Eisenbergiella massiliensis","[Clostridium] spiroforme","Bifidobacterium dentium","Enterocloster aldenensis","Klebsiella pneumoniae","Veillonella infantium","[Eubacterium] sulci","Allisonella histaminiformans","Clostridium bolteae CAG:59","Anaerotignum lactatifermentans","Eubacterium sp. CAG:274","Actinomyces sp. oral taxon 181","Alistipes inops","Bacteroides nordii","Lactobacillus rogosae","Tractidigestivibacter scatoligenes","Bacteroides intestinalis","Fusobacterium nucleatum","Bacteroides salyersiae","Turicimonas muris","Coprococcus eutactus","Firmicutes bacterium CAG:95","Turicibacter sanguinis","Dialister invisus","Desulfovibrio piger","Phocaeicola coprocola","Dielma fastidiosa","Ruminococcaceae bacterium D16","Lacrimispora saccharolytica","Ruminococcaceae bacterium D5","Klebsiella variicola","Dialister pneumosintes","Actinomyces sp. S6-Spd3","Bacteroides sp. CAG:144","Bifidobacterium bifidum","Christensenella hongkongensis","Veillonella sp. T11011-6","Coprobacter secundus","Blautia hydrogenotrophica","Streptococcus sp. A12","Clostridium disporicum","Bacteroides clarus","Harryflintia acetispora","Megamonas funiformis","Acidaminococcus intestini","Tyzzerella nexilis","Christensenella minuta","Eubacterium sp. CAG:251","Haemophilus sp. HMSC71H05","[Clostridium] scindens","Megamonas hypermegale","Prevotella stercorea","Lactococcus lactis","Streptococcus mutans","Alistipes onderdonkii","Paraprevotella clara","Actinomyces oris","Catenibacterium mitsuokai","Prevotella sp. CAG:279","Bacteroides galacturonicus","Klebsiella quasipneumoniae","Coprobacillus cateniformis","Lancefieldella parvula","Fusobacterium mortiferum","Actinomyces sp. HPA0247","Actinomyces graevenitzii","Ruminococcus callidus","Faecalitalea cylindroides","Parabacteroides johnsonii","Pseudoflavonifractor sp. An184","Megasphaera micronuciformis","Monoglobus pectinilyticus","Firmicutes bacterium CAG:170","Dorea sp. CAG:317","Porphyromonas asaccharolytica","Bacteroides faecis","Blautia producta","Fretibacterium fastidiosum","Ligilactobacillus salivarius","Desulfovibrionaceae bacterium","Megasphaera elsdenii","Lachnoclostridium sp. An138","Firmicutes bacterium CAG:238","Prevotella sp. CAG:520","Rothia dentocariosa","Clostridium sp. CAG:299","Enterococcus faecalis","Romboutsia ilealis","Roseburia sp. CAG:309","Lancefieldella rimae","Blautia coccoides","Cloacibacillus porcorum","Lachnoclostridium sp. An131","Streptococcus cristatus","[Eubacterium] brachy","Lactobacillus paragasseri","Victivallis vadensis","Clostridium sp. CAG:167","[Clostridium] methylpentosum","Fusobacterium ulcerans","Bifidobacterium breve","Eubacterium limosum","Oribacterium sinus","Sanguibacteroides justesenii","Butyrivibrio crossotus","Streptococcus sp. F0442","Faecalicoccus pleomorphus","Enterococcus faecium","Massilimicrobiota timonensis","Eubacteriaceae bacterium CHKCI005","Prevotella intermedia","Mitsuokella multacida","Phascolarctobacterium sp. CAG:266","Amedibacillus dolichus","Eubacterium dolichum CAG:375","Peptostreptococcus anaerobius","Fusobacterium periodonticum","Pseudoflavonifractor capillosus","Porphyromonas uenonis","Streptococcus milleri","Limosilactobacillus fermentum","Eubacterium sp. OM08-24","Fusobacterium varium","Blautia hansenii","Bifidobacterium animalis","Bifidobacterium catenulatum","Lactonifactor longoviformis","Bacteroides faecis CAG:32","Scardovia wiggsiae","Alloscardovia omnicolens","Dialister sp. CAG:357","Mogibacterium timidum","Citrobacter youngae","Streptococcus pasteurianus","Campylobacter concisus","Phocaeicola coprophilus","Veillonella rogosae","Aeriscardovia aeriphila","Clostridium sp. CAG:253","Oxalobacter formigenes","Lactobacillus delbrueckii","Fructilactobacillus sanfranciscensis","Clostridium perfringens","Slackia exigua","Parabacteroides gordonii","Granulicatella adiacens","Citrobacter freundii","Schaalia turicensis","Clostridium paraputrificum","Clostridiales bacterium 1_7_47FAA","Parabacteroides sp. CAG:409","Mitsuokella jalaludinii","[Eubacterium] infirmum","Fusobacterium sp. oral taxon 370","Denitrobacterium detoxificans","Butyrivibrio sp. CAG:318","Aggregatibacter segnis","Veillonella tobetsuensis","Roseburia sp. CAG:182","Oribacterium parvum","Oscillibacter sp. PC13","Anaerococcus vaginalis","Flavonifractor sp. An10","Klebsiella oxytoca","Limosilactobacillus mucosae","Prevotella sp. CAG:1185","Actinomyces naeslundii","Alistipes timonensis","Prevotella sp. CAG:5226","Raoultella ornithinolytica","Roseburia sp. CAG:303","Lactobacillus gasseri","Eubacterium callanderi","Brachyspira sp. CAG:700","Morganella morganii","Eikenella corrodens","Olsenella uli","Streptococcus sobrinus","Ruminococcus sp. CAG:330","Prevotella sp. 885","Klebsiella michiganensis","Alloprevotella tannerae","Klebsiella aerogenes","Anaerostipes caccae","Prevotella nigrescens","Pyramidobacter piscolens","Neisseria flavescens","Ruminococcus obeum CAG:39","[Clostridium] hylemonae","Flavonifractor sp. An100","Prevotella sp. AM42-24","Ruminococcus sp. CAG:403","Acidaminococcus fermentans","Lactobacillus crispatus","Prevotella bivia","Butyricicoccus pullicaecorum","Schaalia cardiffensis","Porphyromonas endodontalis","Succinatimonas hippei","Bacteroides sp. CAG:530","Candidatus Stoquefichus sp. KLE1796","Lachnoclostridium sp. An118","Leuconostoc mesenteroides","Enterococcus avium","Dialister succinatiphilus","Lacticaseibacillus rhamnosus","Prevotella sp. CAG:1092","Ruminococcus sp. CAG:488","Odoribacter laneus","Campylobacter ureolyticus","Prevotella melaninogenica","Selenomonas sputigena","Abiotrophia sp. HMSC24B09","Latilactobacillus sakei","Lachnospiraceae bacterium oral taxon 096","Megamonas funiformis CAG:377","Sharpea azabuensis","Prevotella sp. CAG:873","Corynebacterium durum","Prevotella sp. CAG:1031","Megasphaera sp. DISK 18","Anaerofustis stercorihominis","Citrobacter braakii","Clostridium sp. CAG:413","Streptococcus lutetiensis","Streptococcus agalactiae","Lachnoanaerobaculum saburreum","Alloprevotella sp. oral taxon 473","Fusobacterium gonidiaformans","Firmicutes bacterium CAG:424","Abiotrophia defectiva","Clostridium celatum","Mogibacterium pumilum","Campylobacter gracilis","Clostridium sp. CAG:510","Fusobacterium necrophorum","[Eubacterium] nodatum","Bacteroides sp. D2","Propionibacterium freudenreichii","Actinomyces sp. oral taxon 448","Parascardovia denticolens","Enterococcus durans","Prevotella histicola","Prevotella corporis","Gemella asaccharolytica","Prevotella buccae","Filifactor alocis","Parvimonas sp. KA00067","Stomatobaculum longum","Acidaminococcus sp. CAG:542","Actinobaculum sp. oral taxon 183","Actinomyces johnsonii","Anaerofilum sp. An201","Propionibacterium acidifaciens","Prevotella timonensis","Actinomyces viscosus","Limosilactobacillus vaginalis","Hafnia alvei","Ruminococcus sp. CAG:579","Gemmiger sp. An87","Lactococcus petauri","Haemophilus sputorum","Clostridium sp. D5","Streptococcus sp. HPH0090","Methanosphaera stadtmanae","Bacteroides sp. CAG:443","Raoultella planticola","[Eubacterium] saphenum","Prevotella denticola","Citrobacter portucalensis","Bacteroides sp. OM08-11","[Bacteroides] pectinophilus","Anaeroglobus geminatus","Lactococcus garvieae","Actinomyces sp. oral taxon 180","Streptococcus viridans","Enterococcus gallinarum","Limosilactobacillus reuteri","Peptoniphilus harei","Enterobacter mori","Haemophilus haemolyticus","Latilactobacillus curvatus","Atopobium deltae","Actinomyces sp. oral taxon 414","Fusobacterium naviforme","Prevotella oris","Pyramidobacter sp. C12-8","Prevotella pallens","Bacteroides sp. OM05-12","Bacteroides faecichinchillae","Lachnoclostridium sp. An14","Streptococcus sp. M334","Prevotella disiens","Prevotella sp. CAG:891","Limosilactobacillus oris","Lachnospiraceae bacterium 2_1_46FAA","Firmicutes bacterium CAG:646","Streptococcus gallolyticus","Proteus mirabilis","Comamonas kerstersii","Saccharomyces cerevisiae","Clostridium butyricum","Blastocystis sp. subtype 1","[Butyribacterium] methylotrophicum","Streptococcus downei","Candidatus Gastranaerophilales bacterium","Ligilactobacillus ruminis","Enterococcus casseliflavus","Erysipelothrix larvae","Peptoniphilus lacrimalis","Citrobacter pasteurii","Haemophilus parahaemolyticus","Bacteroides stercorirosoris","Hafnia paralvei","Clostridium sp. CAG:632","Corynebacterium argentoratense","Porphyromonas gingivalis","Prevotella sp. CAG:485","Megasphaera sp. MJR8396C","Firmicutes bacterium CAG:534","Anaerotruncus sp. CAG:528","Alistipes sp. An31A","Prevotella buccalis","Eggerthia catenaformis","Clostridium sp. CAG:964","Firmicutes bacterium CAG:555","Haemophilus pittmaniae","Lactobacillus amylovorus","Bifidobacterium pseudolongum","Parvimonas sp. oral taxon 393","Bifidobacterium pullorum","Blautia sp. An249","Blautia argi","Citrobacter koseri","Neisseria sicca","Veillonella sp. CAG:933","Weissella confusa","Campylobacter showae","Lactiplantibacillus plantarum","Peptococcus niger","Bacteroides sp. CAG:598","Prevotella sp. CAG:617","Neisseria subflava","Fusobacterium sp. CAG:439","Finegoldia magna","Bacteroides sp. CAG:462","Prevotella sp. oral taxon 306","Streptococcus sinensis","Bacteroides sp. CAG:633","Fusobacterium equinum","Ruminococcus sp. CAG:563","Staphylococcus aureus","Bacteroides sp. 43_108","Clostridium sp. chh4-2","Citrobacter werkmanii","Aggregatibacter aphrophilus","Megasphaera stantonii","Criibacterium bergeronii","Clostridioides difficile","Streptococcus sp. HMSC034E03","Leuconostoc pseudomesenteroides","Actinomyces massiliensis","Prevotella sp. CAG:1320","Actinomyces sp. oral taxon 897","Clostridiales bacterium CHKCI006","Lactobacillus johnsonii","Firmicutes bacterium CAG:791","Bacteroides fluxus","Streptococcus peroris","Alistipes sp. An66","Veillonella rodentium","Bacteroides sp. CAG:927","Ralstonia pickettii","Faecalicatena orotica","Streptococcus pseudopneumoniae","Campylobacter hominis","Rothia aeria","Leuconostoc lactis","Weissella cibaria","Murimonas intestini","Dialister micraerophilus","Veillonella denticariosi","Citrobacter amalonaticus","Enterobacter bugandensis","Eubacterium coprostanoligenes","Parvimonas sp. oral taxon 110","Lawsonella clevelandensis","Bacteroides sp. CAG:661","Megamonas rupellensis","Aeromonas hydrophila","Campylobacter rectus","Aggregatibacter sp. oral taxon 458","Helicobacter cinaedi","Prevotella salivae","Bacteroides oleiciplenus","Haemophilus paraphrohaemolyticus","Parabacteroides acidifaciens","Lactobacillus acidophilus","Streptococcus sp. NLAE-zl-C503","Pediococcus acidilactici","Bifidobacterium angulatum","Ruminococcus sp. CAG:624","Peptostreptococcus sp. MV1","Streptococcus infantarius","Leptotrichia hofstadii","Porphyromonas sp. HMSC065F10","Enterococcus hirae","Selenomonas noxia","Kluyvera georgiana","Leclercia adecarboxylata","Prevotella sp. oral taxon 376","Citrobacter europaeus","Alloprevotella rava","Brachyspira pilosicoli","Fusobacterium hwasookii","Leuconostoc garlicum","Paraclostridium bifermentans","Lautropia mirabilis","Selenomonas sp. oral taxon 126","Cardiobacterium valvarum","Clostridium sp. 7_2_43FAA","Clostridium sp. CAG:590","Treponema berlinense","Leuconostoc carnosum","Pediococcus pentosaceus","Gardnerella vaginalis","Anaerobiospirillum thomasii","Citrobacter farmeri","Prevotella sp. CAG:755","Proteus hauseri","Neisseria mucosa","Peptoniphilus coxii","Clostridium cadaveris","Atopobium minutum","Aeromonas veronii","Treponema denticola","Aeromonas dhakensis","Bacteroides heparinolyticus","Treponema socranskii","Treponema medium","Varibaculum cambriense","Corynebacterium matruchotii","Arachnia propionica","Rikenella microfusus","Clostridium sp. CAG:411","Pauljensenia hongkongensis","Stenotrophomonas maltophilia","Megasphaera hexanoica","Lactobacillus helveticus","Enterococcus dispar","Enterococcus raffinosus","Actinomyces sp. oral taxon 170","Leptotrichia sp. oral taxon 215","Streptococcus pneumoniae","Subdoligranulum variabile","Lactobacillus iners","Streptococcus sp. HMSC070B10","Marvinbryantia formatexigens","Ezakiella coagulans","Aeromonas caviae","Cetobacterium somerae","Desulfovibrio fairfieldensis","Cutibacterium acnes","Prevotella sp. CAG:1124","Porphyromonas somerae","Shuttleworthia satelles","Flavobacterium sp. FEMGT703F","Negativicoccus succinicivorans","Fannyhessea vaginae","Candidatus Methanomassiliicoccus intestinalis","Kluyvera cryocrescens","Serratia marcescens","Synergistes jonesii","Phocaeicola sartorii","Oribacterium asaccharolyticum","Edwardsiella tarda","Anaeromassilibacillus sp. An172","Limosilactobacillus pontis","Weissella viridescens","Eubacterium sp. An11","Proteus vulgaris","Streptococcus sp. HMSC071D03","Scardovia inopinata","Candida albicans","Lactococcus raffinolactis","Sarcina ventriculi","Leptotrichia sp. oral taxon 212","Limosilactobacillus panis","Aeromonas media","Treponema succinifaciens","Prevotella sp. F0091","Barnesiella sp. An22","Dorea sp. D27","Flavonifractor sp. An306","Enterococcus gilvus","Prevotella veroralis","Haemophilus influenzae","Leuconostoc citreum","Streptococcus sp. oral taxon 058","Lactococcus piscium","Anaerococcus lactolyticus","Moraxella catarrhalis","Aeromonas allosaccharophila","Prevotella jejuni","Leptotrichia wadei","Aliarcobacter butzleri","Sneathia vaginalis","Tyzzerella sp. An114","Peptoniphilus sp. HMSC062D09","Treponema maltophilum","Alistipes sp. CAG:831","Neisseria perflava","Gemmiger sp. An50","Methanobrevibacter arboriphilus","Megasphaera cerevisiae","Methanosphaera sp. SHI1033","Terrisporobacter othiniensis","Cronobacter sakazakii","Limosilactobacillus gastricus","Limosilactobacillus antri","Alcaligenes aquatilis","Citrobacter sp. MGH106","Bacillus aerius","Muribaculum intestinale","Levilactobacillus brevis","Lentilactobacillus parabuchneri","Bifidobacterium scardovii","Streptococcus massiliensis","Lactobacillus kalixensis","Prevotella bergensis","Corynebacterium variabile","Desulfovibrio vulgaris","Kosakonia pseudosacchari","Bifidobacterium ruminantium","Pluralibacter gergoviae","Lachnoclostridium sp. An196","Jonquetella anthropi","Selenomonas artemidis","Prevotella dentalis","Prevotella oralis","Oribacterium sp. oral taxon 078","Sediminibacterium sp. FEMGT703S","Gleimia europaea","Prevotella sp. CAG:1058","Carnobacterium maltaromaticum","Robinsoniella sp. RHS","Leuconostoc gelidum","Escherichia albertii","Lactobacillus kefiranofaciens","Catonella morbi","Clostridium sp. CAG:678","Selenomonas sp. oral taxon 920","Kluyvera ascorbata","Klebsiella grimontii","Treponema vincentii","Clostridium neonatale","Prevotella amnii","Streptococcus sp. oral taxon 056","Cellulosilyticum lentocellum","Prevotella multiformis","Prevotella colorans","Prevotella sp. S7-1-8","Bacteroides pyogenes","Paeniclostridium sordellii","Peptacetobacter hiranonis","Lachnoclostridium sp. An298","Gemmiger sp. An194","Corynebacterium accolens","Atlantibacter hermannii","Neisseria sp. oral taxon 014","Shewanella algae","Raoultella terrigena","Neisseria elongata","Clostridium sp. SY8519","Methanosphaera cuniculi","Campylobacter coli","Staphylococcus equorum","Corynebacterium casei","Schaalia georgiae","Escherichia marmotae","Lachnoanaerobaculum umeaense","Cardiobacterium hominis","Rhodococcus erythropolis","Companilactobacillus mindensis","Staphylococcus haemolyticus","Staphylococcus epidermidis","Acinetobacter johnsonii","Carnobacterium divergens","Actinomyces radicidentis","Companilactobacillus versmoldensis","Staphylococcus warneri","Schaalia meyeri","Kingella oralis","Lachnoclostridium sp. An169","Campylobacter jejuni","Brevibacterium aurantiacum","Brevibacterium linens","Escherichia fergusonii","Bacteroides caecimuris","Ligilactobacillus agilis","Providencia rettgeri","Desulfovibrio legallii","Streptococcus salivarius CAG:79","Trueperella bernardiae","Coprobacter sp.","Pseudomonas fragi","Phocaeicola salanitronis","Proteus penneri","Enterococcus thailandicus","Eikenella longinqua","Pseudoramibacter alactolyticus","Lactiplantibacillus pentosus","Pasteurella canis","Comamonas testosteroni","Facklamia hominis","Cutibacterium namnetense","Malassezia restricta","Tannerella sp. oral taxon 808","Bacteroidetes bacterium oral taxon 272","Enterococcus malodoratus","Bavariicoccus seileri","Tychonema bourrellyi","Actinotignum timonense","Mageeibacillus indolicus","Phascolarctobacterium sp.","Gemmiger sp. An120","Moraxella osloensis","Morococcus cerebrosus","Acetivibrio ethanolgignens","Ligilactobacillus murinus","Lactobacillus intestinalis","Duncaniella muris","Vagococcus teuberi","Peptoanaerobacter stomatis","Tannerella forsythia","Streptococcus sp. HMSC067H01","Pseudomonas lundensis","Serratia liquefaciens","Prevotella oulorum","Bacteroidales bacterium KA00251","Dysgonomonas gadei","Peptoniphilus sp. BV3C26","Clostridium baratii","Lactiplantibacillus paraplantarum","Klebsiella variicola CAG:634","Flavonifractor sp. An82","Prevotella multisaccharivorax","Serratia proteamaculans","Treponema sp. OMZ 838","Selenomonas sp. FOBRC6","Selenomonas sp. oral taxon 138","Pseudomonas helleri","Hungatella effluvii","Roseburia sp. 831b","Enterococcus mundtii","Streptococcus macedonicus","Mitsuokella sp. oral taxon 131","Desulfovibrio sp. MES5","Stenotrophomonas rhizophila","Meiothermus silvanus","Thermus scotoductus","Streptococcus sp. SK643","Photobacterium damselae","Neisseria polysaccharea","Dolosigranulum pigrum","Helicobacter pylori","Centipeda periodontii","Escherichia sp. ESNIH1","Lelliottia jeotgali","Corynebacterium simulans","Haemophilus quentini","Lactobacillus equicursoris","Campylobacter curvus","[Eubacterium] yurii","Methanobrevibacter oralis","Selenomonas infelix","Salmonella enterica","Pseudomonas simiae","Parabacteroides chinchillae","Anaerococcus tetradius","Lactococcus plantarum","Bifidobacterium mongoliense","Anaerosporobacter mobilis","Latilactobacillus fuchuensis","Eremococcus coleocola","Corynebacterium jeikeium","Corynebacterium aurimucosum","Corynebacterium amycolatum","Desulfotomaculum ruminis","Leptotrichia sp. oral taxon 225","Peptoniphilus grossensis","Corynebacterium urealyticum","Methanobacterium sp. MB1","Methanobacterium formicicum","Methanomassiliicoccales archaeon RumEn M1","Haemophilus sp. C1","Clostridium sp. MSTE9","Anaerocolumna aminovalerica","Candida tropicalis"],[0.93,0.93,0.91,0.9,0.88,0.88,0.86,0.86,0.85,0.85,0.83,0.82,0.81,0.8,0.79,0.79,0.78,0.78,0.78,0.76,0.76,0.75,0.75,0.74,0.74,0.74,0.74,0.73,0.73,0.72,0.7,0.69,0.69,0.68,0.68,0.68,0.67,0.67,0.66,0.65,0.64,0.63,0.63,0.62,0.62,0.62,0.62,0.61,0.6,0.59,0.59,0.57,0.56,0.56,0.54,0.54,0.53,0.53,0.53,0.53,0.52,0.52,0.51,0.51,0.5,0.5,0.5,0.5,0.5,0.49,0.48,0.48,0.46,0.46,0.46,0.45,0.45,0.45,0.45,0.43,0.43,0.43,0.42,0.42,0.42,0.41,0.4,0.4,0.4,0.39,0.39,0.39,0.38,0.38,0.38,0.37,0.37,0.36,0.35,0.35,0.34,0.34,0.34,0.34,0.34,0.33,0.33,0.33,0.33,0.33,0.33,0.32,0.32,0.31,0.31,0.31,0.3,0.3,0.3,0.3,0.29,0.29,0.29,0.29,0.29,0.28,0.28,0.28,0.28,0.28,0.27,0.27,0.27,0.27,0.27,0.27,0.27,0.26,0.26,0.25,0.25,0.25,0.24,0.24,0.24,0.24,0.24,0.23,0.23,0.23,0.23,0.23,0.22,0.22,0.22,0.22,0.22,0.21,0.21,0.21,0.21,0.21,0.21,0.21,0.2,0.2,0.2,0.2,0.2,0.2,0.19,0.19,0.19,0.19,0.19,0.19,0.19,0.19,0.19,0.18,0.18,0.18,0.18,0.18,0.17,0.17,0.17,0.17,0.17,0.17,0.17,0.17,0.17,0.16,0.16,0.16,0.15,0.15,0.15,0.15,0.15,0.15,0.15,0.14,0.14,0.14,0.14,0.13,0.13,0.13,0.13,0.13,0.13,0.12,0.12,0.12,0.11,0.11,0.11,0.11,0.11,0.11,0.11,0.11,0.11,0.1,0.1,0.1,0.1,0.098,0.098,0.097,0.093,0.093,0.093,0.091,0.09,0.088,0.088,0.087,0.087,0.087,0.087,0.087,0.085,0.085,0.085,0.085,0.085,0.084,0.083,0.078,0.077,0.077,0.077,0.077,0.075,0.075,0.075,0.075,0.074,0.074,0.073,0.073,0.073,0.073,0.073,0.071,0.07,0.07,0.07,0.07,0.068,0.068,0.068,0.068,0.067,0.066,0.066,0.064,0.063,0.063,0.061,0.061,0.061,0.061,0.06,0.06,0.06,0.058,0.058,0.058,0.058,0.057,0.057,0.057,0.057,0.057,0.056,0.056,0.056,0.056,0.054,0.054,0.054,0.054,0.053,0.053,0.051,0.051,0.051,0.051,0.051,0.051,0.051,0.05,0.05,0.05,0.05,0.048,0.048,0.048,0.048,0.047,0.047,0.046,0.046,0.046,0.044,0.044,0.044,0.043,0.043,0.041,0.04,0.04,0.04,0.038,0.038,0.038,0.038,0.038,0.038,0.038,0.037,0.037,0.036,0.034,0.034,0.034,0.034,0.034,0.033,0.033,0.033,0.033,0.033,0.033,0.033,0.033,0.031,0.031,0.031,0.031,0.031,0.031,0.031,0.031,0.031,0.03,0.03,0.03,0.03,0.03,0.03,0.03,0.03,0.03,0.03,0.028,0.028,0.028,0.027,0.027,0.027,0.027,0.027,0.027,0.027,0.027,0.027,0.027,0.026,0.026,0.026,0.026,0.026,0.026,0.026,0.026,0.026,0.026,0.026,0.024,0.024,0.024,0.024,0.024,0.024,0.024,0.024,0.024,0.024,0.023,0.023,0.023,0.023,0.023,0.023,0.023,0.023,0.021,0.021,0.021,0.021,0.021,0.021,0.021,0.021,0.021,0.021,0.021,0.02,0.02,0.02,0.02,0.02,0.02,0.02,0.02,0.02,0.02,0.02,0.02,0.02,0.019,0.019,0.019,0.019,0.019,0.019,0.017,0.017,0.017,0.017,0.017,0.016,0.016,0.016,0.016,0.016,0.016,0.016,0.016,0.016,0.016,0.016,0.016,0.016,0.016,0.014,0.014,0.014,0.014,0.014,0.014,0.014,0.014,0.014,0.014,0.013,0.013,0.013,0.013,0.013,0.013,0.013,0.013,0.013,0.013,0.013,0.011,0.011,0.011,0.011,0.011,0.011,0.011,0.011,0.011,0.011,0.011,0.011,0.011,0.011,0.01,0.01,0.01,0.01,0.01,0.01,0.01,0.01,0.01,0.01,0.01,0.01,0.01,0.01,0.01,0.01,0.01,0.01,0.01,0.01,0.01,0.01,0.01,0.01,0.0085,0.0085,0.0085,0.0085,0.0085,0.0085,0.0085,0.0085,0.0085,0.0085,0.0085,0.0085,0.0085,0.0085,0.0085,0.0085,0.0085,0.0085,0.0085,0.0085,0.0085,0.0085,0.0071,0.0071,0.0071,0.0071,0.0071,0.0071,0.0071,0.0071,0.0071,0.0071,0.0071,0.0071,0.0071,0.0071,0.0071,0.0071,0.0071,0.0071,0.0071,0.0071,0.0071,0.0071,0.0071,0.0057,0.0057,0.0057,0.0057,0.0057,0.0057,0.0057,0.0057,0.0057,0.0057,0.0057,0.0057,0.0057,0.0057,0.0057,0.0057,0.0057,0.0057,0.0057,0.0057,0.0057,0.0057,0.0057,0.0057,0.0057,0.0057,0.0057,0.0057,0.0057,0.0057,0.0057,0.0057,0.0057,0.0057,0.0043,0.0043,0.0043,0.0043,0.0043,0.0043,0.0043,0.0043,0.0043,0.0043,0.0043,0.0043,0.0043,0.0043,0.0043,0.0043,0.0043,0.0043,0.0043,0.0043,0.0043,0.0043,0.0043,0.0043,0.0043,0.0043,0.0043,0.0043,0.0043,0.0043,0.0043,0.0043,0.0043,0.0043,0.0043,0.0043,0.0043,0.0043,0.0043,0.0043,0.0028,0.0028,0.0028,0.0028,0.0028,0.0028,0.0028,0.0028,0.0028,0.0028,0.0028,0.0028,0.0028,0.0028,0.0028,0.0028,0.0028,0.0028,0.0028,0.0028,0.0028,0.0028,0.0028,0.0028,0.0028,0.0028,0.0028,0.0028,0.0028,0.0028,0.0028,0.0028,0.0028,0.0028,0.0028,0.0028,0.0028,0.0028,0.0028,0.0028,0.0028,0.0028,0.0028,0.0028,0.0028,0.0028,0.0028,0.0028,0.0028,0.0028,0.0028,0.0028,0.0028,0.0028,0.0028,0.0028,0.0028,0.0028,0.0028,0.0028,0.0014,0.0014,0.0014,0.0014,0.0014,0.0014,0.0014,0.0014,0.0014,0.0014,0.0014,0.0014,0.0014,0.0014,0.0014,0.0014,0.0014,0.0014,0.0014,0.0014,0.0014,0.0014,0.0014,0.0014,0.0014,0.0014,0.0014,0.0014,0.0014,0.0014,0.0014,0.0014,0.0014,0.0014,0.0014,0.0014,0.0014,0.0014,0.0014,0.0014,0.0014,0.0014,0.0014,0.0014,0.0014,0.0014,0.0014,0.0014,0.0014,0.0014,0.0014,0.0014,0.0014,0.0014,0.0014,0.0014,0.0014,0.0014,0.0014,0.0014,0.0014,0.0014,0.0014,0.0014,0.0014,0.0014,0.0014,0.0014,0.0014,0.0014,0.0014,0.0014,0.0014,0.0014,0.0014,0.0014,0.0014,0.0014,0.0014,0.0014,0.0014,0.0014,0.0014,0.0014,0.0014,0.0014,0.0014,0.0014,0.0014,0.0014,0.0014,0.0014,0.0014,0.0014,0.0014,0.0014,0.0014,0.0014,0.0014,0.0014,0.0014,0.0014,0.0014,0.0014,0.0014,0.0014,0.0014,0.0014,0.0014,0.0014,0.0014,0.0014,0.0014,0.0014,0.0014,0.0014,0.0014,0.0014,0.0014,0.0014,0.0014,0.0014,0.0014]],"container":"<table class=\"display\">\n  <thead>\n    <tr>\n      <th> <\/th>\n      <th>species<\/th>\n      <th>prevalence<\/th>\n    <\/tr>\n  <\/thead>\n<\/table>","options":{"columnDefs":[{"className":"dt-right","targets":2},{"orderable":false,"targets":0}],"order":[],"autoWidth":false,"orderClasses":false}},"evals":[],"jsHooks":[]}</script><p>Write to disk:</p>
<div class="sourceCode" id="cb5"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="https://rdrr.io/r/utils/write.table.html" class="external-link">write.csv</a></span><span class="op">(</span><span class="va">prevalences</span>, row.names <span class="op">=</span> <span class="cn">FALSE</span>, file <span class="op">=</span> <span class="st">"prevalences.csv"</span><span class="op">)</span></code></pre></div>
<p>Download the zipped prevalences file produced by <em><a href="https://bioconductor.org/packages/3.14/curatedMetagenomicData" class="external-link">curatedMetagenomicData</a></em>
version 3.2.3: <a href="https://github.com/waldronlab/curatedMetagenomicAnalyses/files/8254646/prevalences.csv.zip">prevalences.csv.zip</a></p>
</div>
<div class="section level2">
<h2 id="session-info">Session Info<a class="anchor" aria-label="anchor" href="#session-info"></a>
</h2>
<div class="sourceCode" id="cb6"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu">sessioninfo</span><span class="fu">::</span><span class="fu"><a href="https://r-lib.github.io/sessioninfo/reference/session_info.html" class="external-link">session_info</a></span><span class="op">(</span><span class="op">)</span></code></pre></div>
<pre><code><span class="co">## <span style="color: #00BBBB; font-weight: bold;">â”€ Session info â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€</span></span>
<span class="co">##  <span style="color: #555555; font-style: italic;">setting </span> <span style="color: #555555; font-style: italic;">value</span></span>
<span class="co">##  version  R version 4.1.3 (2022-03-10)</span>
<span class="co">##  os       Ubuntu 20.04.4 LTS</span>
<span class="co">##  system   x86_64, linux-gnu</span>
<span class="co">##  ui       X11</span>
<span class="co">##  language en</span>
<span class="co">##  collate  en_US.UTF-8</span>
<span class="co">##  ctype    en_US.UTF-8</span>
<span class="co">##  tz       UTC</span>
<span class="co">##  date     2022-04-06</span>
<span class="co">##  pandoc   2.17.1.1 @ /usr/local/bin/ (via rmarkdown)</span>
<span class="co">## </span>
<span class="co">## <span style="color: #00BBBB; font-weight: bold;">â”€ Packages â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€</span></span>
<span class="co">##  <span style="color: #555555; font-style: italic;">package                 </span> <span style="color: #555555; font-style: italic;">*</span> <span style="color: #555555; font-style: italic;">version </span> <span style="color: #555555; font-style: italic;">date (UTC)</span> <span style="color: #555555; font-style: italic;">lib</span> <span style="color: #555555; font-style: italic;">source</span></span>
<span class="co">##  AnnotationDbi              1.56.2   <span style="color: #555555;">2021-11-09</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">Bioconductor</span></span>
<span class="co">##  AnnotationHub              3.2.2    <span style="color: #555555;">2022-03-01</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">Bioconductor</span></span>
<span class="co">##  ape                        5.6-2    <span style="color: #555555;">2022-03-02</span> <span style="color: #555555;">[1]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM (R 4.1.0)</span></span>
<span class="co">##  assertthat                 0.2.1    <span style="color: #555555;">2019-03-21</span> <span style="color: #555555;">[1]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM (R 4.1.0)</span></span>
<span class="co">##  beachmat                   2.10.0   <span style="color: #555555;">2021-10-26</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">Bioconductor</span></span>
<span class="co">##  beeswarm                   0.4.0    <span style="color: #555555;">2021-06-01</span> <span style="color: #555555;">[1]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM (R 4.1.0)</span></span>
<span class="co">##  Biobase                  * 2.54.0   <span style="color: #555555;">2021-10-26</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">Bioconductor</span></span>
<span class="co">##  BiocFileCache              2.2.1    <span style="color: #555555;">2022-01-23</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">Bioconductor</span></span>
<span class="co">##  BiocGenerics             * 0.40.0   <span style="color: #555555;">2021-10-26</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">Bioconductor</span></span>
<span class="co">##  BiocManager                1.30.16  <span style="color: #555555;">2021-06-15</span> <span style="color: #555555;">[2]</span> <span style="color: #555555;">CRAN (R 4.1.3)</span></span>
<span class="co">##  BiocNeighbors              1.12.0   <span style="color: #555555;">2021-10-26</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">Bioconductor</span></span>
<span class="co">##  BiocParallel               1.28.3   <span style="color: #555555;">2021-12-09</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">Bioconductor</span></span>
<span class="co">##  BiocSingular               1.10.0   <span style="color: #555555;">2021-10-26</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">Bioconductor</span></span>
<span class="co">##  BiocStyle                * 2.22.0   <span style="color: #555555;">2021-10-26</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">Bioconductor</span></span>
<span class="co">##  BiocVersion                3.14.0   <span style="color: #555555;">2021-05-19</span> <span style="color: #555555;">[2]</span> <span style="color: #555555;">Bioconductor</span></span>
<span class="co">##  Biostrings               * 2.62.0   <span style="color: #555555;">2021-10-26</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">Bioconductor</span></span>
<span class="co">##  bit                        4.0.4    <span style="color: #555555;">2020-08-04</span> <span style="color: #555555;">[1]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM (R 4.1.0)</span></span>
<span class="co">##  bit64                      4.0.5    <span style="color: #555555;">2020-08-30</span> <span style="color: #555555;">[1]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM (R 4.1.0)</span></span>
<span class="co">##  bitops                     1.0-7    <span style="color: #555555;">2021-04-24</span> <span style="color: #555555;">[1]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM (R 4.1.0)</span></span>
<span class="co">##  blob                       1.2.2    <span style="color: #555555;">2021-07-23</span> <span style="color: #555555;">[1]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM (R 4.1.0)</span></span>
<span class="co">##  bookdown                   0.25     <span style="color: #555555;">2022-03-16</span> <span style="color: #555555;">[1]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM (R 4.1.0)</span></span>
<span class="co">##  bslib                      0.3.1    <span style="color: #555555;">2021-10-06</span> <span style="color: #555555;">[1]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM (R 4.1.0)</span></span>
<span class="co">##  cachem                     1.0.6    <span style="color: #555555;">2021-08-19</span> <span style="color: #555555;">[2]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM (R 4.1.0)</span></span>
<span class="co">##  cli                        3.2.0    <span style="color: #555555;">2022-02-14</span> <span style="color: #555555;">[2]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM (R 4.1.0)</span></span>
<span class="co">##  cluster                    2.1.3    <span style="color: #555555;">2022-03-28</span> <span style="color: #555555;">[3]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM (R 4.1.0)</span></span>
<span class="co">##  colorspace                 2.0-3    <span style="color: #555555;">2022-02-21</span> <span style="color: #555555;">[1]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM (R 4.1.0)</span></span>
<span class="co">##  crayon                     1.5.1    <span style="color: #555555;">2022-03-26</span> <span style="color: #555555;">[2]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM (R 4.1.0)</span></span>
<span class="co">##  crosstalk                  1.2.0    <span style="color: #555555;">2021-11-04</span> <span style="color: #555555;">[1]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM (R 4.1.0)</span></span>
<span class="co">##  curatedMetagenomicData   * 3.2.3    <span style="color: #555555;">2021-12-22</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">Bioconductor</span></span>
<span class="co">##  curl                       4.3.2    <span style="color: #555555;">2021-06-23</span> <span style="color: #555555;">[2]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM (R 4.1.0)</span></span>
<span class="co">##  DBI                        1.1.2    <span style="color: #555555;">2021-12-20</span> <span style="color: #555555;">[1]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM (R 4.1.0)</span></span>
<span class="co">##  dbplyr                     2.1.1    <span style="color: #555555;">2021-04-06</span> <span style="color: #555555;">[1]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM (R 4.1.0)</span></span>
<span class="co">##  DECIPHER                   2.22.0   <span style="color: #555555;">2021-10-26</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">Bioconductor</span></span>
<span class="co">##  decontam                   1.14.0   <span style="color: #555555;">2021-10-26</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">Bioconductor</span></span>
<span class="co">##  DelayedArray               0.20.0   <span style="color: #555555;">2021-10-26</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">Bioconductor</span></span>
<span class="co">##  DelayedMatrixStats         1.16.0   <span style="color: #555555;">2021-10-26</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">Bioconductor</span></span>
<span class="co">##  desc                       1.4.1    <span style="color: #555555;">2022-03-06</span> <span style="color: #555555;">[2]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM (R 4.1.0)</span></span>
<span class="co">##  digest                     0.6.29   <span style="color: #555555;">2021-12-01</span> <span style="color: #555555;">[2]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM (R 4.1.0)</span></span>
<span class="co">##  DirichletMultinomial       1.36.0   <span style="color: #555555;">2021-10-26</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">Bioconductor</span></span>
<span class="co">##  dplyr                    * 1.0.8    <span style="color: #555555;">2022-02-08</span> <span style="color: #555555;">[2]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM (R 4.1.0)</span></span>
<span class="co">##  DT                       * 0.22     <span style="color: #555555;">2022-03-28</span> <span style="color: #555555;">[1]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM (R 4.1.0)</span></span>
<span class="co">##  ellipsis                   0.3.2    <span style="color: #555555;">2021-04-29</span> <span style="color: #555555;">[2]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM (R 4.1.0)</span></span>
<span class="co">##  evaluate                   0.15     <span style="color: #555555;">2022-02-18</span> <span style="color: #555555;">[2]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM (R 4.1.0)</span></span>
<span class="co">##  ExperimentHub              2.2.1    <span style="color: #555555;">2022-01-23</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">Bioconductor</span></span>
<span class="co">##  fansi                      1.0.3    <span style="color: #555555;">2022-03-24</span> <span style="color: #555555;">[2]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM (R 4.1.0)</span></span>
<span class="co">##  fastmap                    1.1.0    <span style="color: #555555;">2021-01-25</span> <span style="color: #555555;">[2]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM (R 4.1.0)</span></span>
<span class="co">##  filelock                   1.0.2    <span style="color: #555555;">2018-10-05</span> <span style="color: #555555;">[1]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM (R 4.1.0)</span></span>
<span class="co">##  Formula                    1.2-4    <span style="color: #555555;">2020-10-16</span> <span style="color: #555555;">[1]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM (R 4.1.0)</span></span>
<span class="co">##  fs                         1.5.2    <span style="color: #555555;">2021-12-08</span> <span style="color: #555555;">[2]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM (R 4.1.0)</span></span>
<span class="co">##  generics                   0.1.2    <span style="color: #555555;">2022-01-31</span> <span style="color: #555555;">[2]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM (R 4.1.0)</span></span>
<span class="co">##  GenomeInfoDb             * 1.30.1   <span style="color: #555555;">2022-01-30</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">Bioconductor</span></span>
<span class="co">##  GenomeInfoDbData           1.2.7    <span style="color: #555555;">2022-04-06</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">Bioconductor</span></span>
<span class="co">##  GenomicRanges            * 1.46.1   <span style="color: #555555;">2021-11-18</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">Bioconductor</span></span>
<span class="co">##  ggbeeswarm                 0.6.0    <span style="color: #555555;">2017-08-07</span> <span style="color: #555555;">[1]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM (R 4.1.0)</span></span>
<span class="co">##  ggplot2                    3.3.5    <span style="color: #555555;">2021-06-25</span> <span style="color: #555555;">[1]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM (R 4.1.0)</span></span>
<span class="co">##  ggrepel                    0.9.1    <span style="color: #555555;">2021-01-15</span> <span style="color: #555555;">[1]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM (R 4.1.0)</span></span>
<span class="co">##  glue                       1.6.2    <span style="color: #555555;">2022-02-24</span> <span style="color: #555555;">[2]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM (R 4.1.0)</span></span>
<span class="co">##  gridExtra                  2.3      <span style="color: #555555;">2017-09-09</span> <span style="color: #555555;">[1]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM (R 4.1.0)</span></span>
<span class="co">##  gtable                     0.3.0    <span style="color: #555555;">2019-03-25</span> <span style="color: #555555;">[1]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM (R 4.1.0)</span></span>
<span class="co">##  htmltools                  0.5.2    <span style="color: #555555;">2021-08-25</span> <span style="color: #555555;">[1]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM (R 4.1.0)</span></span>
<span class="co">##  htmlwidgets                1.5.4    <span style="color: #555555;">2021-09-08</span> <span style="color: #555555;">[1]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM (R 4.1.0)</span></span>
<span class="co">##  httpuv                     1.6.5    <span style="color: #555555;">2022-01-05</span> <span style="color: #555555;">[1]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM (R 4.1.0)</span></span>
<span class="co">##  httr                       1.4.2    <span style="color: #555555;">2020-07-20</span> <span style="color: #555555;">[2]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM (R 4.1.0)</span></span>
<span class="co">##  interactiveDisplayBase     1.32.0   <span style="color: #555555;">2021-10-26</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">Bioconductor</span></span>
<span class="co">##  IRanges                  * 2.28.0   <span style="color: #555555;">2021-10-26</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">Bioconductor</span></span>
<span class="co">##  irlba                      2.3.5    <span style="color: #555555;">2021-12-06</span> <span style="color: #555555;">[1]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM (R 4.1.0)</span></span>
<span class="co">##  jquerylib                  0.1.4    <span style="color: #555555;">2021-04-26</span> <span style="color: #555555;">[1]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM (R 4.1.0)</span></span>
<span class="co">##  jsonlite                   1.8.0    <span style="color: #555555;">2022-02-22</span> <span style="color: #555555;">[2]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM (R 4.1.0)</span></span>
<span class="co">##  KEGGREST                   1.34.0   <span style="color: #555555;">2021-10-26</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">Bioconductor</span></span>
<span class="co">##  knitr                      1.38     <span style="color: #555555;">2022-03-25</span> <span style="color: #555555;">[2]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM (R 4.1.0)</span></span>
<span class="co">##  later                      1.3.0    <span style="color: #555555;">2021-08-18</span> <span style="color: #555555;">[1]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM (R 4.1.0)</span></span>
<span class="co">##  lattice                    0.20-45  <span style="color: #555555;">2021-09-22</span> <span style="color: #555555;">[3]</span> <span style="color: #555555;">CRAN (R 4.1.3)</span></span>
<span class="co">##  lazyeval                   0.2.2    <span style="color: #555555;">2019-03-15</span> <span style="color: #555555;">[1]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM (R 4.1.0)</span></span>
<span class="co">##  lifecycle                  1.0.1    <span style="color: #555555;">2021-09-24</span> <span style="color: #555555;">[2]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM (R 4.1.0)</span></span>
<span class="co">##  magrittr                   2.0.3    <span style="color: #555555;">2022-03-30</span> <span style="color: #555555;">[2]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM (R 4.1.0)</span></span>
<span class="co">##  MASS                       7.3-56   <span style="color: #555555;">2022-03-23</span> <span style="color: #555555;">[3]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM (R 4.1.0)</span></span>
<span class="co">##  Matrix                     1.4-1    <span style="color: #555555;">2022-03-23</span> <span style="color: #555555;">[3]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM (R 4.1.0)</span></span>
<span class="co">##  MatrixGenerics           * 1.6.0    <span style="color: #555555;">2021-10-26</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">Bioconductor</span></span>
<span class="co">##  matrixStats              * 0.61.0   <span style="color: #555555;">2021-09-17</span> <span style="color: #555555;">[1]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM (R 4.1.0)</span></span>
<span class="co">##  memoise                    2.0.1    <span style="color: #555555;">2021-11-26</span> <span style="color: #555555;">[2]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM (R 4.1.0)</span></span>
<span class="co">##  mgcv                       1.8-40   <span style="color: #555555;">2022-03-29</span> <span style="color: #555555;">[3]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM (R 4.1.0)</span></span>
<span class="co">##  mia                        1.2.7    <span style="color: #555555;">2022-02-08</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">Bioconductor</span></span>
<span class="co">##  mime                       0.12     <span style="color: #555555;">2021-09-28</span> <span style="color: #555555;">[2]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM (R 4.1.0)</span></span>
<span class="co">##  MultiAssayExperiment       1.20.0   <span style="color: #555555;">2021-10-26</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">Bioconductor</span></span>
<span class="co">##  munsell                    0.5.0    <span style="color: #555555;">2018-06-12</span> <span style="color: #555555;">[1]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM (R 4.1.0)</span></span>
<span class="co">##  nlme                       3.1-157  <span style="color: #555555;">2022-03-25</span> <span style="color: #555555;">[3]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM (R 4.1.0)</span></span>
<span class="co">##  permute                    0.9-7    <span style="color: #555555;">2022-01-27</span> <span style="color: #555555;">[1]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM (R 4.1.0)</span></span>
<span class="co">##  pillar                     1.7.0    <span style="color: #555555;">2022-02-01</span> <span style="color: #555555;">[2]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM (R 4.1.0)</span></span>
<span class="co">##  pkgconfig                  2.0.3    <span style="color: #555555;">2019-09-22</span> <span style="color: #555555;">[2]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM (R 4.1.0)</span></span>
<span class="co">##  pkgdown                    2.0.2    <span style="color: #555555;">2022-01-13</span> <span style="color: #555555;">[1]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM (R 4.1.0)</span></span>
<span class="co">##  plyr                       1.8.7    <span style="color: #555555;">2022-03-24</span> <span style="color: #555555;">[1]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM (R 4.1.0)</span></span>
<span class="co">##  png                        0.1-7    <span style="color: #555555;">2013-12-03</span> <span style="color: #555555;">[1]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM (R 4.1.0)</span></span>
<span class="co">##  promises                   1.2.0.1  <span style="color: #555555;">2021-02-11</span> <span style="color: #555555;">[1]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM (R 4.1.0)</span></span>
<span class="co">##  purrr                      0.3.4    <span style="color: #555555;">2020-04-17</span> <span style="color: #555555;">[2]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM (R 4.1.0)</span></span>
<span class="co">##  R6                         2.5.1    <span style="color: #555555;">2021-08-19</span> <span style="color: #555555;">[2]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM (R 4.1.0)</span></span>
<span class="co">##  ragg                       1.2.2    <span style="color: #555555;">2022-02-21</span> <span style="color: #555555;">[1]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM (R 4.1.0)</span></span>
<span class="co">##  rappdirs                   0.3.3    <span style="color: #555555;">2021-01-31</span> <span style="color: #555555;">[2]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM (R 4.1.0)</span></span>
<span class="co">##  Rcpp                       1.0.8.3  <span style="color: #555555;">2022-03-17</span> <span style="color: #555555;">[1]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM (R 4.1.0)</span></span>
<span class="co">##  RCurl                      1.98-1.6 <span style="color: #555555;">2022-02-08</span> <span style="color: #555555;">[1]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM (R 4.1.0)</span></span>
<span class="co">##  reshape2                   1.4.4    <span style="color: #555555;">2020-04-09</span> <span style="color: #555555;">[1]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM (R 4.1.0)</span></span>
<span class="co">##  rlang                      1.0.2    <span style="color: #555555;">2022-03-04</span> <span style="color: #555555;">[2]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM (R 4.1.0)</span></span>
<span class="co">##  rmarkdown                  2.13     <span style="color: #555555;">2022-03-10</span> <span style="color: #555555;">[1]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM (R 4.1.0)</span></span>
<span class="co">##  rprojroot                  2.0.3    <span style="color: #555555;">2022-04-02</span> <span style="color: #555555;">[2]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM (R 4.1.0)</span></span>
<span class="co">##  RSQLite                    2.2.12   <span style="color: #555555;">2022-04-02</span> <span style="color: #555555;">[1]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM (R 4.1.0)</span></span>
<span class="co">##  rsvd                       1.0.5    <span style="color: #555555;">2021-04-16</span> <span style="color: #555555;">[1]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM (R 4.1.0)</span></span>
<span class="co">##  S4Vectors                * 0.32.4   <span style="color: #555555;">2022-03-24</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">Bioconductor</span></span>
<span class="co">##  sass                       0.4.1    <span style="color: #555555;">2022-03-23</span> <span style="color: #555555;">[1]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM (R 4.1.0)</span></span>
<span class="co">##  ScaledMatrix               1.2.0    <span style="color: #555555;">2021-10-26</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">Bioconductor</span></span>
<span class="co">##  scales                     1.1.1    <span style="color: #555555;">2020-05-11</span> <span style="color: #555555;">[1]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM (R 4.1.0)</span></span>
<span class="co">##  scater                     1.22.0   <span style="color: #555555;">2021-10-26</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">Bioconductor</span></span>
<span class="co">##  scuttle                    1.4.0    <span style="color: #555555;">2021-10-26</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">Bioconductor</span></span>
<span class="co">##  sessioninfo                1.2.2    <span style="color: #555555;">2021-12-06</span> <span style="color: #555555;">[2]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM (R 4.1.0)</span></span>
<span class="co">##  shiny                      1.7.1    <span style="color: #555555;">2021-10-02</span> <span style="color: #555555;">[1]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM (R 4.1.0)</span></span>
<span class="co">##  SingleCellExperiment     * 1.16.0   <span style="color: #555555;">2021-10-26</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">Bioconductor</span></span>
<span class="co">##  sparseMatrixStats          1.6.0    <span style="color: #555555;">2021-10-26</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">Bioconductor</span></span>
<span class="co">##  stringi                    1.7.6    <span style="color: #555555;">2021-11-29</span> <span style="color: #555555;">[2]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM (R 4.1.0)</span></span>
<span class="co">##  stringr                    1.4.0    <span style="color: #555555;">2019-02-10</span> <span style="color: #555555;">[2]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM (R 4.1.0)</span></span>
<span class="co">##  SummarizedExperiment     * 1.24.0   <span style="color: #555555;">2021-10-26</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">Bioconductor</span></span>
<span class="co">##  systemfonts                1.0.4    <span style="color: #555555;">2022-02-11</span> <span style="color: #555555;">[1]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM (R 4.1.0)</span></span>
<span class="co">##  table1                   * 1.4.2    <span style="color: #555555;">2021-06-06</span> <span style="color: #555555;">[1]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM (R 4.1.0)</span></span>
<span class="co">##  textshaping                0.3.6    <span style="color: #555555;">2021-10-13</span> <span style="color: #555555;">[1]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM (R 4.1.0)</span></span>
<span class="co">##  tibble                     3.1.6    <span style="color: #555555;">2021-11-07</span> <span style="color: #555555;">[2]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM (R 4.1.0)</span></span>
<span class="co">##  tidyr                      1.2.0    <span style="color: #555555;">2022-02-01</span> <span style="color: #555555;">[2]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM (R 4.1.0)</span></span>
<span class="co">##  tidyselect                 1.1.2    <span style="color: #555555;">2022-02-21</span> <span style="color: #555555;">[2]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM (R 4.1.0)</span></span>
<span class="co">##  tidytree                   0.3.9    <span style="color: #555555;">2022-03-04</span> <span style="color: #555555;">[1]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM (R 4.1.0)</span></span>
<span class="co">##  treeio                     1.18.1   <span style="color: #555555;">2021-11-14</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">Bioconductor</span></span>
<span class="co">##  TreeSummarizedExperiment * 2.2.0    <span style="color: #555555;">2021-10-26</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">Bioconductor</span></span>
<span class="co">##  utf8                       1.2.2    <span style="color: #555555;">2021-07-24</span> <span style="color: #555555;">[2]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM (R 4.1.0)</span></span>
<span class="co">##  vctrs                      0.4.0    <span style="color: #555555;">2022-03-30</span> <span style="color: #555555;">[2]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM (R 4.1.0)</span></span>
<span class="co">##  vegan                      2.5-7    <span style="color: #555555;">2020-11-28</span> <span style="color: #555555;">[1]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM (R 4.1.0)</span></span>
<span class="co">##  vipor                      0.4.5    <span style="color: #555555;">2017-03-22</span> <span style="color: #555555;">[1]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM (R 4.1.0)</span></span>
<span class="co">##  viridis                    0.6.2    <span style="color: #555555;">2021-10-13</span> <span style="color: #555555;">[1]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM (R 4.1.0)</span></span>
<span class="co">##  viridisLite                0.4.0    <span style="color: #555555;">2021-04-13</span> <span style="color: #555555;">[1]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM (R 4.1.0)</span></span>
<span class="co">##  withr                      2.5.0    <span style="color: #555555;">2022-03-03</span> <span style="color: #555555;">[2]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM (R 4.1.0)</span></span>
<span class="co">##  xfun                       0.30     <span style="color: #555555;">2022-03-02</span> <span style="color: #555555;">[2]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM (R 4.1.0)</span></span>
<span class="co">##  xtable                     1.8-4    <span style="color: #555555;">2019-04-21</span> <span style="color: #555555;">[1]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM (R 4.1.0)</span></span>
<span class="co">##  XVector                  * 0.34.0   <span style="color: #555555;">2021-10-26</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">Bioconductor</span></span>
<span class="co">##  yaml                       2.3.5    <span style="color: #555555;">2022-02-21</span> <span style="color: #555555;">[2]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM (R 4.1.0)</span></span>
<span class="co">##  yulab.utils                0.0.4    <span style="color: #555555;">2021-10-09</span> <span style="color: #555555;">[1]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM (R 4.1.0)</span></span>
<span class="co">##  zlibbioc                   1.40.0   <span style="color: #555555;">2021-10-26</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">Bioconductor</span></span>
<span class="co">## </span>
<span class="co">## <span style="color: #555555;"> [1] /__w/_temp/Library</span></span>
<span class="co">## <span style="color: #555555;"> [2] /usr/local/lib/R/site-library</span></span>
<span class="co">## <span style="color: #555555;"> [3] /usr/local/lib/R/library</span></span>
<span class="co">## </span>
<span class="co">## <span style="color: #00BBBB; font-weight: bold;">â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€</span></span></code></pre>
</div>
  </div>

  <div class="col-md-3 hidden-xs hidden-sm" id="pkgdown-sidebar">

        <nav id="toc" data-toggle="toc"><h2 data-toc-skip>Contents</h2>
    </nav>
</div>

</div>



      <footer><div class="copyright">
  <p></p>
<p>Developed by Levi Waldron, Paolo Manghi, Davide Galzato, Lucas
Schiffer, Jennifer Wokaty.</p>
</div>

<div class="pkgdown">
  <p></p>
<p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a>
2.0.2.</p>
</div>

      </footer>
</div>

  


  

  </body>
</html>
